// vue-dark-switch/dist/exports.mjs
import "./exports.css";
import { isVNode as At, Comment as Ft, Fragment as Et, watch as _e, computed as F, ref as K, onMounted as xr, readonly as br, inject as de, provide as Wt, onBeforeUnmount as Lt, onBeforeMount as vr, watchEffect as $o, defineComponent as te, h as I, Transition as Sr, toRef as yr, mergeProps as Rt, isRef as $r, unref as N, getCurrentInstance as jt, nextTick as Pr, getCurrentScope as Vt, onScopeDispose as Nt, shallowRef as Gt, openBlock as me, createElementBlock as wr, createElementVNode as Hr, createBlock as je, withCtx as ie, createVNode as ge, renderSlot as Ne } from "vue";
function X(e) {
  return typeof e == "string" ? e.endsWith("px") ? Number(e.slice(0, e.length - 2)) : Number(e) : e;
}
function so(e) {
  if (e != null)
    return typeof e == "number" ? `${e}px` : e.endsWith("px") ? e : `${e}px`;
}
const Oo = {
  black: "#000",
  silver: "#C0C0C0",
  gray: "#808080",
  white: "#FFF",
  maroon: "#800000",
  red: "#F00",
  purple: "#800080",
  fuchsia: "#F0F",
  green: "#008000",
  lime: "#0F0",
  olive: "#808000",
  yellow: "#FF0",
  navy: "#000080",
  blue: "#00F",
  teal: "#008080",
  aqua: "#0FF",
  transparent: "#0000"
}, xe = "^\\s*", be = "\\s*$", ae = "\\s*((\\.\\d+)|(\\d+(\\.\\d*)?))\\s*", le = "([0-9A-Fa-f])", se = "([0-9A-Fa-f]{2})", Ut = new RegExp(`${xe}rgb\\s*\\(${ae},${ae},${ae}\\)${be}`), qt = new RegExp(`${xe}rgba\\s*\\(${ae},${ae},${ae},${ae}\\)${be}`), Kt = new RegExp(`${xe}#${le}${le}${le}${be}`), Qt = new RegExp(`${xe}#${se}${se}${se}${be}`), Jt = new RegExp(`${xe}#${le}${le}${le}${le}${be}`), Yt = new RegExp(`${xe}#${se}${se}${se}${se}${be}`);
function V(e) {
  return parseInt(e, 16);
}
function Z(e) {
  try {
    let o;
    if (o = Qt.exec(e))
      return [V(o[1]), V(o[2]), V(o[3]), 1];
    if (o = Ut.exec(e))
      return [R(o[1]), R(o[5]), R(o[9]), 1];
    if (o = qt.exec(e))
      return [
        R(o[1]),
        R(o[5]),
        R(o[9]),
        ze(o[13])
      ];
    if (o = Kt.exec(e))
      return [
        V(o[1] + o[1]),
        V(o[2] + o[2]),
        V(o[3] + o[3]),
        1
      ];
    if (o = Yt.exec(e))
      return [
        V(o[1]),
        V(o[2]),
        V(o[3]),
        ze(V(o[4]) / 255)
      ];
    if (o = Jt.exec(e))
      return [
        V(o[1] + o[1]),
        V(o[2] + o[2]),
        V(o[3] + o[3]),
        ze(V(o[4] + o[4]) / 255)
      ];
    if (e in Oo)
      return Z(Oo[e]);
    throw new Error(`[seemly/rgba]: Invalid color value ${e}.`);
  } catch (o) {
    throw o;
  }
}
function Xt(e) {
  return e > 1 ? 1 : e < 0 ? 0 : e;
}
function go(e, o, r, t) {
  return `rgba(${R(e)}, ${R(o)}, ${R(r)}, ${Xt(t)})`;
}
function co(e, o, r, t, n) {
  return R((e * o * (1 - t) + r * t) / n);
}
function P(e, o) {
  Array.isArray(e) || (e = Z(e)), Array.isArray(o) || (o = Z(o));
  const r = e[3], t = o[3], n = ze(r + t - r * t);
  return go(co(e[0], r, o[0], t, n), co(e[1], r, o[1], t, n), co(e[2], r, o[2], t, n), n);
}
function S(e, o) {
  const [r, t, n, i = 1] = Array.isArray(e) ? e : Z(e);
  return o.alpha ? go(r, t, n, o.alpha) : go(r, t, n, i);
}
function A(e, o) {
  const [r, t, n, i = 1] = Array.isArray(e) ? e : Z(e), { lightness: a = 1, alpha: l = 1 } = o;
  return Zt([r * a, t * a, n * a, i * l]);
}
function ze(e) {
  const o = Math.round(Number(e) * 100) / 100;
  return o > 1 ? 1 : o < 0 ? 0 : o;
}
function R(e) {
  const o = Math.round(Number(e));
  return o > 255 ? 255 : o < 0 ? 0 : o;
}
function Zt(e) {
  const [o, r, t] = e;
  return 3 in e ? `rgba(${R(o)}, ${R(r)}, ${R(t)}, ${ze(e[3])})` : `rgba(${R(o)}, ${R(r)}, ${R(t)}, 1)`;
}
function Ve(e, ...o) {
  if (Array.isArray(e))
    e.forEach((r) => Ve(r, ...o));
  else
    return e(...o);
}
const Bo = /* @__PURE__ */ new Set();
function en(e, o) {
  const r = `[naive/${e}]: ${o}`;
  Bo.has(r) || (Bo.add(r), console.error(r));
}
function on(e, o) {
  console.error(`[naive/${e}]: ${o}`);
}
function Dr(e, o) {
  throw new Error(`[naive/${e}]: ${o}`);
}
function Po(e) {
  return e.some((o) => At(o) ? !(o.type === Ft || o.type === Et && !Po(o.children)) : !0) ? e : null;
}
function ne(e, o) {
  const r = e && Po(e());
  return o(r || null);
}
function po(e) {
  return !(e && Po(e()));
}
const rn = /^(\d|\.)+$/, Ao = /(\d|\.)+/;
function tn(e, { c: o = 1, offset: r = 0, attachPx: t = !0 } = {}) {
  if (typeof e == "number") {
    const n = (e + r) * o;
    return n === 0 ? "0" : `${n}px`;
  } else if (typeof e == "string")
    if (rn.test(e)) {
      const n = (Number(e) + r) * o;
      return t ? n === 0 ? "0" : `${n}px` : `${n}`;
    } else {
      const n = Ao.exec(e);
      return n ? e.replace(Ao, String((Number(n[0]) + r) * o)) : e;
    }
  return e;
}
function nn(e) {
  let o = 0;
  for (let r = 0; r < e.length; ++r)
    e[r] === "&" && ++o;
  return o;
}
const Tr = /\s*,(?![^(]*\))\s*/g, an = /\s+/g;
function ln(e, o) {
  const r = [];
  return o.split(Tr).forEach((t) => {
    let n = nn(t);
    if (n) {
      if (n === 1) {
        e.forEach((a) => {
          r.push(t.replace("&", a));
        });
        return;
      }
    } else {
      e.forEach((a) => {
        r.push(
          // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
          (a && a + " ") + t
        );
      });
      return;
    }
    let i = [
      t
    ];
    for (; n--; ) {
      const a = [];
      i.forEach((l) => {
        e.forEach((s) => {
          a.push(l.replace("&", s));
        });
      }), i = a;
    }
    i.forEach((a) => r.push(a));
  }), r;
}
function sn(e, o) {
  const r = [];
  return o.split(Tr).forEach((t) => {
    e.forEach((n) => {
      r.push((n && n + " ") + t);
    });
  }), r;
}
function cn(e) {
  let o = [""];
  return e.forEach((r) => {
    r = r && r.trim(), // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
    r && (r.includes("&") ? o = ln(o, r) : o = sn(o, r));
  }), o.join(", ").replace(an, " ");
}
function Fo(e) {
  if (!e)
    return;
  const o = e.parentElement;
  o && o.removeChild(e);
}
function wo(e) {
  return document.querySelector(`style[cssr-id="${e}"]`);
}
function dn(e) {
  const o = document.createElement("style");
  return o.setAttribute("cssr-id", e), o;
}
function We(e) {
  return e ? /^\s*@(s|m)/.test(e) : !1;
}
const pn = /[A-Z]/g;
function Ir(e) {
  return e.replace(pn, (o) => "-" + o.toLowerCase());
}
function un(e, o = "  ") {
  return typeof e == "object" && e !== null ? ` {
` + Object.entries(e).map((r) => o + `  ${Ir(r[0])}: ${r[1]};`).join(`
`) + `
` + o + "}" : `: ${e};`;
}
function fn(e, o, r) {
  return typeof e == "function" ? e({
    context: o.context,
    props: r
  }) : e;
}
function Eo(e, o, r, t) {
  if (!o)
    return "";
  const n = fn(o, r, t);
  if (!n)
    return "";
  if (typeof n == "string")
    return `${e} {
${n}
}`;
  const i = Object.keys(n);
  if (i.length === 0)
    return r.config.keepEmptyBlock ? e + ` {
}` : "";
  const a = e ? [
    e + " {"
  ] : [];
  return i.forEach((l) => {
    const s = n[l];
    if (l === "raw") {
      a.push(`
` + s + `
`);
      return;
    }
    l = Ir(l), s != null && a.push(`  ${l}${un(s)}`);
  }), e && a.push("}"), a.join(`
`);
}
function mo(e, o, r) {
  e && e.forEach((t) => {
    if (Array.isArray(t))
      mo(t, o, r);
    else if (typeof t == "function") {
      const n = t(o);
      Array.isArray(n) ? mo(n, o, r) : n && r(n);
    } else
      t && r(t);
  });
}
function zr(e, o, r, t, n, i) {
  const a = e.$;
  let l = "";
  if (!a || typeof a == "string")
    We(a) ? l = a : o.push(a);
  else if (typeof a == "function") {
    const d = a({
      context: t.context,
      props: n
    });
    We(d) ? l = d : o.push(d);
  } else if (a.before && a.before(t.context), !a.$ || typeof a.$ == "string")
    We(a.$) ? l = a.$ : o.push(a.$);
  else if (a.$) {
    const d = a.$({
      context: t.context,
      props: n
    });
    We(d) ? l = d : o.push(d);
  }
  const s = cn(o), c = Eo(s, e.props, t, n);
  l ? (r.push(`${l} {`), i && c && i.insertRule(`${l} {
${c}
}
`)) : (i && c && i.insertRule(c), !i && c.length && r.push(c)), e.children && mo(e.children, {
    context: t.context,
    props: n
  }, (d) => {
    if (typeof d == "string") {
      const u = Eo(s, { raw: d }, t, n);
      i ? i.insertRule(u) : r.push(u);
    } else
      zr(d, o, r, t, n, i);
  }), o.pop(), l && r.push("}"), a && a.after && a.after(t.context);
}
function kr(e, o, r, t = !1) {
  const n = [];
  return zr(e, [], n, o, r, t ? e.instance.__styleSheet : void 0), t ? "" : n.join(`

`);
}
function Co(e) {
  for (var o = 0, r, t = 0, n = e.length; n >= 4; ++t, n -= 4)
    r = e.charCodeAt(t) & 255 | (e.charCodeAt(++t) & 255) << 8 | (e.charCodeAt(++t) & 255) << 16 | (e.charCodeAt(++t) & 255) << 24, r = /* Math.imul(k, m): */
    (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16), r ^= /* k >>> r: */
    r >>> 24, o = /* Math.imul(k, m): */
    (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16) ^ /* Math.imul(h, m): */
    (o & 65535) * 1540483477 + ((o >>> 16) * 59797 << 16);
  switch (n) {
    case 3:
      o ^= (e.charCodeAt(t + 2) & 255) << 16;
    case 2:
      o ^= (e.charCodeAt(t + 1) & 255) << 8;
    case 1:
      o ^= e.charCodeAt(t) & 255, o = /* Math.imul(h, m): */
      (o & 65535) * 1540483477 + ((o >>> 16) * 59797 << 16);
  }
  return o ^= o >>> 13, o = /* Math.imul(h, m): */
  (o & 65535) * 1540483477 + ((o >>> 16) * 59797 << 16), ((o ^ o >>> 15) >>> 0).toString(36);
}
typeof window < "u" && (window.__cssrContext = {});
function gn(e, o, r) {
  const { els: t } = o;
  if (r === void 0)
    t.forEach(Fo), o.els = [];
  else {
    const n = wo(r);
    n && t.includes(n) && (Fo(n), o.els = t.filter((i) => i !== n));
  }
}
function Wo(e, o) {
  e.push(o);
}
function mn(e, o, r, t, n, i, a, l, s) {
  if (i && !s) {
    if (r === void 0) {
      console.error("[css-render/mount]: `id` is required in `silent` mode.");
      return;
    }
    const g = window.__cssrContext;
    g[r] || (g[r] = !0, kr(o, e, t, i));
    return;
  }
  let c;
  if (r === void 0 && (c = o.render(t), r = Co(c)), s) {
    s.adapter(r, c ?? o.render(t));
    return;
  }
  const d = wo(r);
  if (d !== null && !a)
    return d;
  const u = d ?? dn(r);
  if (c === void 0 && (c = o.render(t)), u.textContent = c, d !== null)
    return d;
  if (l) {
    const g = document.head.querySelector(`meta[name="${l}"]`);
    if (g)
      return document.head.insertBefore(u, g), Wo(o.els, u), u;
  }
  return n ? document.head.insertBefore(u, document.head.querySelector("style, link")) : document.head.appendChild(u), Wo(o.els, u), u;
}
function Cn(e) {
  return kr(this, this.instance, e);
}
function hn(e = {}) {
  const { id: o, ssr: r, props: t, head: n = !1, silent: i = !1, force: a = !1, anchorMetaName: l } = e;
  return mn(this.instance, this, o, t, n, i, a, l, r);
}
function xn(e = {}) {
  const { id: o } = e;
  gn(this.instance, this, o);
}
const Le = function(e, o, r, t) {
  return {
    instance: e,
    $: o,
    props: r,
    children: t,
    els: [],
    render: Cn,
    mount: hn,
    unmount: xn
  };
}, bn = function(e, o, r, t) {
  return Array.isArray(o) ? Le(e, { $: null }, null, o) : Array.isArray(r) ? Le(e, o, null, r) : Array.isArray(t) ? Le(e, o, r, t) : Le(e, o, r, null);
};
function vn(e = {}) {
  let o = null;
  const r = {
    c: (...t) => bn(r, ...t),
    use: (t, ...n) => t.install(r, ...n),
    find: wo,
    context: {},
    config: e,
    get __styleSheet() {
      if (!o) {
        const t = document.createElement("style");
        return document.head.appendChild(t), o = document.styleSheets[document.styleSheets.length - 1], o;
      }
      return o;
    }
  };
  return r;
}
function Sn(e) {
  let o = ".", r = "__", t = "--", n;
  if (e) {
    let p = e.blockPrefix;
    p && (o = p), p = e.elementPrefix, p && (r = p), p = e.modifierPrefix, p && (t = p);
  }
  const i = {
    install(p) {
      n = p.c;
      const h = p.context;
      h.bem = {}, h.bem.b = null, h.bem.els = null;
    }
  };
  function a(p) {
    let h, x;
    return {
      before(m) {
        h = m.bem.b, x = m.bem.els, m.bem.els = null;
      },
      after(m) {
        m.bem.b = h, m.bem.els = x;
      },
      $({ context: m, props: y }) {
        return p = typeof p == "string" ? p : p({ context: m, props: y }), m.bem.b = p, `${(y == null ? void 0 : y.bPrefix) || o}${m.bem.b}`;
      }
    };
  }
  function l(p) {
    let h;
    return {
      before(x) {
        h = x.bem.els;
      },
      after(x) {
        x.bem.els = h;
      },
      $({ context: x, props: m }) {
        return p = typeof p == "string" ? p : p({ context: x, props: m }), x.bem.els = p.split(",").map((y) => y.trim()), x.bem.els.map((y) => `${(m == null ? void 0 : m.bPrefix) || o}${x.bem.b}${r}${y}`).join(", ");
      }
    };
  }
  function s(p) {
    return {
      $({ context: h, props: x }) {
        p = typeof p == "string" ? p : p({ context: h, props: x });
        const m = p.split(",").map((z) => z.trim());
        function y(z) {
          return m.map((_) => `&${(x == null ? void 0 : x.bPrefix) || o}${h.bem.b}${z !== void 0 ? `${r}${z}` : ""}${t}${_}`).join(", ");
        }
        const w = h.bem.els;
        if (w !== null) {
          if (process.env.NODE_ENV !== "production" && w.length >= 2)
            throw Error(`[css-render/plugin-bem]: m(${p}) is invalid, using modifier inside multiple elements is not allowed`);
          return y(w[0]);
        } else
          return y();
      }
    };
  }
  function c(p) {
    return {
      $({ context: h, props: x }) {
        p = typeof p == "string" ? p : p({ context: h, props: x });
        const m = h.bem.els;
        if (process.env.NODE_ENV !== "production" && m !== null && m.length >= 2)
          throw Error(`[css-render/plugin-bem]: notM(${p}) is invalid, using modifier inside multiple elements is not allowed`);
        return `&:not(${(x == null ? void 0 : x.bPrefix) || o}${h.bem.b}${m !== null && m.length > 0 ? `${r}${m[0]}` : ""}${t}${p})`;
      }
    };
  }
  return Object.assign(i, {
    cB: (...p) => n(a(p[0]), p[1], p[2]),
    cE: (...p) => n(l(p[0]), p[1], p[2]),
    cM: (...p) => n(s(p[0]), p[1], p[2]),
    cNotM: (...p) => n(c(p[0]), p[1], p[2])
  }), i;
}
function re(e, o) {
  return e + (o === "default" ? "" : o.replace(/^[a-z]/, (r) => r.toUpperCase()));
}
re("abc", "def");
const yn = "n", $n = `.${yn}-`, Pn = "__", wn = "--", _r = vn(), Mr = Sn({
  blockPrefix: $n,
  elementPrefix: Pn,
  modifierPrefix: wn
});
_r.use(Mr);
const { c: j, find: mf } = _r, { cB: Ge, cE: D, cM: J, cNotM: Lo } = Mr;
function Hn(e, o) {
  return _e(e, (r) => {
    r !== void 0 && (o.value = r);
  }), F(() => e.value === void 0 ? o.value : e.value);
}
function Dn() {
  const e = K(!1);
  return xr(() => {
    e.value = !0;
  }), br(e);
}
const Or = Symbol("@css-render/vue3-ssr");
function Tn(e, o) {
  return `<style cssr-id="${e}">
${o}
</style>`;
}
function In(e, o) {
  const r = de(Or, null);
  if (r === null) {
    console.error("[css-render/vue3-ssr]: no ssr context found.");
    return;
  }
  const { styles: t, ids: n } = r;
  n.has(e) || t !== null && (n.add(e), t.push(Tn(e, o)));
}
const zn = typeof document < "u";
function Ho() {
  if (zn)
    return;
  const e = de(Or, null);
  if (e !== null)
    return {
      adapter: In,
      context: e
    };
}
const Ro = "n-form-item";
function kn(e, { defaultSize: o = "medium", mergedSize: r, mergedDisabled: t } = {}) {
  const n = de(Ro, null);
  Wt(Ro, null);
  const i = F(r ? () => r(n) : () => {
    const { size: s } = e;
    if (s)
      return s;
    if (n) {
      const { mergedSize: c } = n;
      if (c.value !== void 0)
        return c.value;
    }
    return o;
  }), a = F(t ? () => t(n) : () => {
    const { disabled: s } = e;
    return s !== void 0 ? s : n ? n.disabled.value : !1;
  }), l = F(() => {
    const { status: s } = e;
    return s || (n == null ? void 0 : n.mergedValidationStatus.value);
  });
  return Lt(() => {
    n && n.restoreValidation();
  }), {
    mergedSizeRef: i,
    mergedDisabledRef: a,
    mergedStatusRef: l,
    nTriggerFormBlur() {
      n && n.handleContentBlur();
    },
    nTriggerFormChange() {
      n && n.handleContentChange();
    },
    nTriggerFormFocus() {
      n && n.handleContentFocus();
    },
    nTriggerFormInput() {
      n && n.handleContentInput();
    }
  };
}
var _n = typeof global == "object" && global && global.Object === Object && global;
const Br = _n;
var Mn = typeof self == "object" && self && self.Object === Object && self, On = Br || Mn || Function("return this")();
const ve = On;
var Bn = ve.Symbol;
const Ue = Bn;
var Ar = Object.prototype, An = Ar.hasOwnProperty, Fn = Ar.toString, Te = Ue ? Ue.toStringTag : void 0;
function En(e) {
  var o = An.call(e, Te), r = e[Te];
  try {
    e[Te] = void 0;
    var t = !0;
  } catch {
  }
  var n = Fn.call(e);
  return t && (o ? e[Te] = r : delete e[Te]), n;
}
var Wn = Object.prototype, Ln = Wn.toString;
function Rn(e) {
  return Ln.call(e);
}
var jn = "[object Null]", Vn = "[object Undefined]", jo = Ue ? Ue.toStringTag : void 0;
function Xe(e) {
  return e == null ? e === void 0 ? Vn : jn : jo && jo in Object(e) ? En(e) : Rn(e);
}
function Be(e) {
  return e != null && typeof e == "object";
}
var Nn = Array.isArray;
const ho = Nn;
function pe(e) {
  var o = typeof e;
  return e != null && (o == "object" || o == "function");
}
function Fr(e) {
  return e;
}
var Gn = "[object AsyncFunction]", Un = "[object Function]", qn = "[object GeneratorFunction]", Kn = "[object Proxy]";
function Do(e) {
  if (!pe(e))
    return !1;
  var o = Xe(e);
  return o == Un || o == qn || o == Gn || o == Kn;
}
var Qn = ve["__core-js_shared__"];
const uo = Qn;
var Vo = function() {
  var e = /[^.]+$/.exec(uo && uo.keys && uo.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Jn(e) {
  return !!Vo && Vo in e;
}
var Yn = Function.prototype, Xn = Yn.toString;
function Zn(e) {
  if (e != null) {
    try {
      return Xn.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var ei = /[\\^$.*+?()[\]{}|]/g, oi = /^\[object .+?Constructor\]$/, ri = Function.prototype, ti = Object.prototype, ni = ri.toString, ii = ti.hasOwnProperty, ai = RegExp(
  "^" + ni.call(ii).replace(ei, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function li(e) {
  if (!pe(e) || Jn(e))
    return !1;
  var o = Do(e) ? ai : oi;
  return o.test(Zn(e));
}
function si(e, o) {
  return e == null ? void 0 : e[o];
}
function To(e, o) {
  var r = si(e, o);
  return li(r) ? r : void 0;
}
var No = Object.create, ci = function() {
  function e() {
  }
  return function(o) {
    if (!pe(o))
      return {};
    if (No)
      return No(o);
    e.prototype = o;
    var r = new e();
    return e.prototype = void 0, r;
  };
}();
const di = ci;
function pi(e, o, r) {
  switch (r.length) {
    case 0:
      return e.call(o);
    case 1:
      return e.call(o, r[0]);
    case 2:
      return e.call(o, r[0], r[1]);
    case 3:
      return e.call(o, r[0], r[1], r[2]);
  }
  return e.apply(o, r);
}
function ui(e, o) {
  var r = -1, t = e.length;
  for (o || (o = Array(t)); ++r < t; )
    o[r] = e[r];
  return o;
}
var fi = 800, gi = 16, mi = Date.now;
function Ci(e) {
  var o = 0, r = 0;
  return function() {
    var t = mi(), n = gi - (t - r);
    if (r = t, n > 0) {
      if (++o >= fi)
        return arguments[0];
    } else
      o = 0;
    return e.apply(void 0, arguments);
  };
}
function hi(e) {
  return function() {
    return e;
  };
}
var xi = function() {
  try {
    var e = To(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const qe = xi;
var bi = qe ? function(e, o) {
  return qe(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: hi(o),
    writable: !0
  });
} : Fr;
const vi = bi;
var Si = Ci(vi);
const yi = Si;
var $i = 9007199254740991, Pi = /^(?:0|[1-9]\d*)$/;
function Er(e, o) {
  var r = typeof e;
  return o = o ?? $i, !!o && (r == "number" || r != "symbol" && Pi.test(e)) && e > -1 && e % 1 == 0 && e < o;
}
function Io(e, o, r) {
  o == "__proto__" && qe ? qe(e, o, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[o] = r;
}
function Ze(e, o) {
  return e === o || e !== e && o !== o;
}
var wi = Object.prototype, Hi = wi.hasOwnProperty;
function Di(e, o, r) {
  var t = e[o];
  (!(Hi.call(e, o) && Ze(t, r)) || r === void 0 && !(o in e)) && Io(e, o, r);
}
function Ti(e, o, r, t) {
  var n = !r;
  r || (r = {});
  for (var i = -1, a = o.length; ++i < a; ) {
    var l = o[i], s = t ? t(r[l], e[l], l, r, e) : void 0;
    s === void 0 && (s = e[l]), n ? Io(r, l, s) : Di(r, l, s);
  }
  return r;
}
var Go = Math.max;
function Ii(e, o, r) {
  return o = Go(o === void 0 ? e.length - 1 : o, 0), function() {
    for (var t = arguments, n = -1, i = Go(t.length - o, 0), a = Array(i); ++n < i; )
      a[n] = t[o + n];
    n = -1;
    for (var l = Array(o + 1); ++n < o; )
      l[n] = t[n];
    return l[o] = r(a), pi(e, this, l);
  };
}
function zi(e, o) {
  return yi(Ii(e, o, Fr), e + "");
}
var ki = 9007199254740991;
function Wr(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= ki;
}
function zo(e) {
  return e != null && Wr(e.length) && !Do(e);
}
function _i(e, o, r) {
  if (!pe(r))
    return !1;
  var t = typeof o;
  return (t == "number" ? zo(r) && Er(o, r.length) : t == "string" && o in r) ? Ze(r[o], e) : !1;
}
function Mi(e) {
  return zi(function(o, r) {
    var t = -1, n = r.length, i = n > 1 ? r[n - 1] : void 0, a = n > 2 ? r[2] : void 0;
    for (i = e.length > 3 && typeof i == "function" ? (n--, i) : void 0, a && _i(r[0], r[1], a) && (i = n < 3 ? void 0 : i, n = 1), o = Object(o); ++t < n; ) {
      var l = r[t];
      l && e(o, l, t, i);
    }
    return o;
  });
}
var Oi = Object.prototype;
function Lr(e) {
  var o = e && e.constructor, r = typeof o == "function" && o.prototype || Oi;
  return e === r;
}
function Bi(e, o) {
  for (var r = -1, t = Array(e); ++r < e; )
    t[r] = o(r);
  return t;
}
var Ai = "[object Arguments]";
function Uo(e) {
  return Be(e) && Xe(e) == Ai;
}
var Rr = Object.prototype, Fi = Rr.hasOwnProperty, Ei = Rr.propertyIsEnumerable, Wi = Uo(function() {
  return arguments;
}()) ? Uo : function(e) {
  return Be(e) && Fi.call(e, "callee") && !Ei.call(e, "callee");
};
const xo = Wi;
function Li() {
  return !1;
}
var jr = typeof exports == "object" && exports && !exports.nodeType && exports, qo = jr && typeof module == "object" && module && !module.nodeType && module, Ri = qo && qo.exports === jr, Ko = Ri ? ve.Buffer : void 0, ji = Ko ? Ko.isBuffer : void 0, Vi = ji || Li;
const Vr = Vi;
var Ni = "[object Arguments]", Gi = "[object Array]", Ui = "[object Boolean]", qi = "[object Date]", Ki = "[object Error]", Qi = "[object Function]", Ji = "[object Map]", Yi = "[object Number]", Xi = "[object Object]", Zi = "[object RegExp]", ea = "[object Set]", oa = "[object String]", ra = "[object WeakMap]", ta = "[object ArrayBuffer]", na = "[object DataView]", ia = "[object Float32Array]", aa = "[object Float64Array]", la = "[object Int8Array]", sa = "[object Int16Array]", ca = "[object Int32Array]", da = "[object Uint8Array]", pa = "[object Uint8ClampedArray]", ua = "[object Uint16Array]", fa = "[object Uint32Array]", O = {};
O[ia] = O[aa] = O[la] = O[sa] = O[ca] = O[da] = O[pa] = O[ua] = O[fa] = !0;
O[Ni] = O[Gi] = O[ta] = O[Ui] = O[na] = O[qi] = O[Ki] = O[Qi] = O[Ji] = O[Yi] = O[Xi] = O[Zi] = O[ea] = O[oa] = O[ra] = !1;
function ga(e) {
  return Be(e) && Wr(e.length) && !!O[Xe(e)];
}
function ma(e) {
  return function(o) {
    return e(o);
  };
}
var Nr = typeof exports == "object" && exports && !exports.nodeType && exports, ke = Nr && typeof module == "object" && module && !module.nodeType && module, Ca = ke && ke.exports === Nr, fo = Ca && Br.process, ha = function() {
  try {
    var e = ke && ke.require && ke.require("util").types;
    return e || fo && fo.binding && fo.binding("util");
  } catch {
  }
}();
const Qo = ha;
var Jo = Qo && Qo.isTypedArray, xa = Jo ? ma(Jo) : ga;
const Gr = xa;
var ba = Object.prototype, va = ba.hasOwnProperty;
function Sa(e, o) {
  var r = ho(e), t = !r && xo(e), n = !r && !t && Vr(e), i = !r && !t && !n && Gr(e), a = r || t || n || i, l = a ? Bi(e.length, String) : [], s = l.length;
  for (var c in e)
    (o || va.call(e, c)) && !(a && // Safari 9 has enumerable `arguments.length` in strict mode.
    (c == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    n && (c == "offset" || c == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    i && (c == "buffer" || c == "byteLength" || c == "byteOffset") || // Skip index properties.
    Er(c, s))) && l.push(c);
  return l;
}
function ya(e, o) {
  return function(r) {
    return e(o(r));
  };
}
function $a(e) {
  var o = [];
  if (e != null)
    for (var r in Object(e))
      o.push(r);
  return o;
}
var Pa = Object.prototype, wa = Pa.hasOwnProperty;
function Ha(e) {
  if (!pe(e))
    return $a(e);
  var o = Lr(e), r = [];
  for (var t in e)
    t == "constructor" && (o || !wa.call(e, t)) || r.push(t);
  return r;
}
function Ur(e) {
  return zo(e) ? Sa(e, !0) : Ha(e);
}
var Da = To(Object, "create");
const Me = Da;
function Ta() {
  this.__data__ = Me ? Me(null) : {}, this.size = 0;
}
function Ia(e) {
  var o = this.has(e) && delete this.__data__[e];
  return this.size -= o ? 1 : 0, o;
}
var za = "__lodash_hash_undefined__", ka = Object.prototype, _a = ka.hasOwnProperty;
function Ma(e) {
  var o = this.__data__;
  if (Me) {
    var r = o[e];
    return r === za ? void 0 : r;
  }
  return _a.call(o, e) ? o[e] : void 0;
}
var Oa = Object.prototype, Ba = Oa.hasOwnProperty;
function Aa(e) {
  var o = this.__data__;
  return Me ? o[e] !== void 0 : Ba.call(o, e);
}
var Fa = "__lodash_hash_undefined__";
function Ea(e, o) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = Me && o === void 0 ? Fa : o, this;
}
function ce(e) {
  var o = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++o < r; ) {
    var t = e[o];
    this.set(t[0], t[1]);
  }
}
ce.prototype.clear = Ta;
ce.prototype.delete = Ia;
ce.prototype.get = Ma;
ce.prototype.has = Aa;
ce.prototype.set = Ea;
function Wa() {
  this.__data__ = [], this.size = 0;
}
function eo(e, o) {
  for (var r = e.length; r--; )
    if (Ze(e[r][0], o))
      return r;
  return -1;
}
var La = Array.prototype, Ra = La.splice;
function ja(e) {
  var o = this.__data__, r = eo(o, e);
  if (r < 0)
    return !1;
  var t = o.length - 1;
  return r == t ? o.pop() : Ra.call(o, r, 1), --this.size, !0;
}
function Va(e) {
  var o = this.__data__, r = eo(o, e);
  return r < 0 ? void 0 : o[r][1];
}
function Na(e) {
  return eo(this.__data__, e) > -1;
}
function Ga(e, o) {
  var r = this.__data__, t = eo(r, e);
  return t < 0 ? (++this.size, r.push([e, o])) : r[t][1] = o, this;
}
function ee(e) {
  var o = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++o < r; ) {
    var t = e[o];
    this.set(t[0], t[1]);
  }
}
ee.prototype.clear = Wa;
ee.prototype.delete = ja;
ee.prototype.get = Va;
ee.prototype.has = Na;
ee.prototype.set = Ga;
var Ua = To(ve, "Map");
const qr = Ua;
function qa() {
  this.size = 0, this.__data__ = {
    hash: new ce(),
    map: new (qr || ee)(),
    string: new ce()
  };
}
function Ka(e) {
  var o = typeof e;
  return o == "string" || o == "number" || o == "symbol" || o == "boolean" ? e !== "__proto__" : e === null;
}
function oo(e, o) {
  var r = e.__data__;
  return Ka(o) ? r[typeof o == "string" ? "string" : "hash"] : r.map;
}
function Qa(e) {
  var o = oo(this, e).delete(e);
  return this.size -= o ? 1 : 0, o;
}
function Ja(e) {
  return oo(this, e).get(e);
}
function Ya(e) {
  return oo(this, e).has(e);
}
function Xa(e, o) {
  var r = oo(this, e), t = r.size;
  return r.set(e, o), this.size += r.size == t ? 0 : 1, this;
}
function Se(e) {
  var o = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++o < r; ) {
    var t = e[o];
    this.set(t[0], t[1]);
  }
}
Se.prototype.clear = qa;
Se.prototype.delete = Qa;
Se.prototype.get = Ja;
Se.prototype.has = Ya;
Se.prototype.set = Xa;
var Za = ya(Object.getPrototypeOf, Object);
const Kr = Za;
var el = "[object Object]", ol = Function.prototype, rl = Object.prototype, Qr = ol.toString, tl = rl.hasOwnProperty, nl = Qr.call(Object);
function il(e) {
  if (!Be(e) || Xe(e) != el)
    return !1;
  var o = Kr(e);
  if (o === null)
    return !0;
  var r = tl.call(o, "constructor") && o.constructor;
  return typeof r == "function" && r instanceof r && Qr.call(r) == nl;
}
function al() {
  this.__data__ = new ee(), this.size = 0;
}
function ll(e) {
  var o = this.__data__, r = o.delete(e);
  return this.size = o.size, r;
}
function sl(e) {
  return this.__data__.get(e);
}
function cl(e) {
  return this.__data__.has(e);
}
var dl = 200;
function pl(e, o) {
  var r = this.__data__;
  if (r instanceof ee) {
    var t = r.__data__;
    if (!qr || t.length < dl - 1)
      return t.push([e, o]), this.size = ++r.size, this;
    r = this.__data__ = new Se(t);
  }
  return r.set(e, o), this.size = r.size, this;
}
function ye(e) {
  var o = this.__data__ = new ee(e);
  this.size = o.size;
}
ye.prototype.clear = al;
ye.prototype.delete = ll;
ye.prototype.get = sl;
ye.prototype.has = cl;
ye.prototype.set = pl;
var Jr = typeof exports == "object" && exports && !exports.nodeType && exports, Yo = Jr && typeof module == "object" && module && !module.nodeType && module, ul = Yo && Yo.exports === Jr, Xo = ul ? ve.Buffer : void 0, Zo = Xo ? Xo.allocUnsafe : void 0;
function fl(e, o) {
  if (o)
    return e.slice();
  var r = e.length, t = Zo ? Zo(r) : new e.constructor(r);
  return e.copy(t), t;
}
var gl = ve.Uint8Array;
const er = gl;
function ml(e) {
  var o = new e.constructor(e.byteLength);
  return new er(o).set(new er(e)), o;
}
function Cl(e, o) {
  var r = o ? ml(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
function hl(e) {
  return typeof e.constructor == "function" && !Lr(e) ? di(Kr(e)) : {};
}
function xl(e) {
  return function(o, r, t) {
    for (var n = -1, i = Object(o), a = t(o), l = a.length; l--; ) {
      var s = a[e ? l : ++n];
      if (r(i[s], s, i) === !1)
        break;
    }
    return o;
  };
}
var bl = xl();
const vl = bl;
function bo(e, o, r) {
  (r !== void 0 && !Ze(e[o], r) || r === void 0 && !(o in e)) && Io(e, o, r);
}
function Sl(e) {
  return Be(e) && zo(e);
}
function vo(e, o) {
  if (!(o === "constructor" && typeof e[o] == "function") && o != "__proto__")
    return e[o];
}
function yl(e) {
  return Ti(e, Ur(e));
}
function $l(e, o, r, t, n, i, a) {
  var l = vo(e, r), s = vo(o, r), c = a.get(s);
  if (c) {
    bo(e, r, c);
    return;
  }
  var d = i ? i(l, s, r + "", e, o, a) : void 0, u = d === void 0;
  if (u) {
    var g = ho(s), f = !g && Vr(s), p = !g && !f && Gr(s);
    d = s, g || f || p ? ho(l) ? d = l : Sl(l) ? d = ui(l) : f ? (u = !1, d = fl(s, !0)) : p ? (u = !1, d = Cl(s, !0)) : d = [] : il(s) || xo(s) ? (d = l, xo(l) ? d = yl(l) : (!pe(l) || Do(l)) && (d = hl(s))) : u = !1;
  }
  u && (a.set(s, d), n(d, s, t, i, a), a.delete(s)), bo(e, r, d);
}
function Yr(e, o, r, t, n) {
  e !== o && vl(o, function(i, a) {
    if (n || (n = new ye()), pe(i))
      $l(e, o, a, r, Yr, t, n);
    else {
      var l = t ? t(vo(e, a), i, a + "", e, o, n) : void 0;
      l === void 0 && (l = i), bo(e, a, l);
    }
  }, Ur);
}
var Pl = Mi(function(e, o, r) {
  Yr(e, o, r);
});
const Re = Pl, ro = {
  fontFamily: 'v-sans, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol"',
  fontFamilyMono: "v-mono, SFMono-Regular, Menlo, Consolas, Courier, monospace",
  fontWeight: "400",
  fontWeightStrong: "500",
  cubicBezierEaseInOut: "cubic-bezier(.4, 0, .2, 1)",
  cubicBezierEaseOut: "cubic-bezier(0, 0, .2, 1)",
  cubicBezierEaseIn: "cubic-bezier(.4, 0, 1, 1)",
  borderRadius: "3px",
  borderRadiusSmall: "2px",
  fontSize: "14px",
  fontSizeMini: "12px",
  fontSizeTiny: "12px",
  fontSizeSmall: "14px",
  fontSizeMedium: "14px",
  fontSizeLarge: "15px",
  fontSizeHuge: "16px",
  lineHeight: "1.6",
  heightMini: "16px",
  heightTiny: "22px",
  heightSmall: "28px",
  heightMedium: "34px",
  heightLarge: "40px",
  heightHuge: "46px"
}, {
  fontSize: wl,
  fontFamily: Hl,
  lineHeight: Dl
} = ro, Xr = j("body", `
 margin: 0;
 font-size: ${wl};
 font-family: ${Hl};
 line-height: ${Dl};
 -webkit-text-size-adjust: 100%;
 -webkit-tap-highlight-color: transparent;
`, [j("input", `
 font-family: inherit;
 font-size: inherit;
 `)]), to = "n-config-provider", Ke = "naive-ui-style";
function Ae(e, o, r, t, n, i) {
  const a = Ho(), l = de(to, null);
  if (r) {
    const c = () => {
      const d = i == null ? void 0 : i.value;
      r.mount({
        id: d === void 0 ? o : d + o,
        head: !0,
        props: {
          bPrefix: d ? `.${d}-` : void 0
        },
        anchorMetaName: Ke,
        ssr: a
      }), l != null && l.preflightStyleDisabled || Xr.mount({
        id: "n-global",
        head: !0,
        anchorMetaName: Ke,
        ssr: a
      });
    };
    a ? c() : vr(c);
  }
  return F(() => {
    var c;
    const { theme: { common: d, self: u, peers: g = {} } = {}, themeOverrides: f = {}, builtinThemeOverrides: p = {} } = n, { common: h, peers: x } = f, { common: m = void 0, [e]: { common: y = void 0, self: w = void 0, peers: z = {} } = {} } = (l == null ? void 0 : l.mergedThemeRef.value) || {}, { common: _ = void 0, [e]: B = {} } = (l == null ? void 0 : l.mergedThemeOverridesRef.value) || {}, { common: k, peers: v = {} } = B, H = Re({}, d || y || m || t.common, _, k, h), M = Re(
      // {}, executed every time, no need for empty obj
      (c = u || w || t.self) === null || c === void 0 ? void 0 : c(H),
      p,
      B,
      f
    );
    return {
      common: H,
      self: M,
      peers: Re({}, t.peers, z, g),
      peerOverrides: Re({}, p.peers, v, x)
    };
  });
}
Ae.props = {
  theme: Object,
  themeOverrides: Object,
  builtinThemeOverrides: Object
};
const Tl = "n";
function Zr(e = {}, o = {
  defaultBordered: !0
}) {
  const r = de(to, null);
  return {
    // NConfigProvider,
    inlineThemeDisabled: r == null ? void 0 : r.inlineThemeDisabled,
    mergedRtlRef: r == null ? void 0 : r.mergedRtlRef,
    mergedComponentPropsRef: r == null ? void 0 : r.mergedComponentPropsRef,
    mergedBreakpointsRef: r == null ? void 0 : r.mergedBreakpointsRef,
    mergedBorderedRef: F(() => {
      var t, n;
      const { bordered: i } = e;
      return i !== void 0 ? i : (n = (t = r == null ? void 0 : r.mergedBorderedRef.value) !== null && t !== void 0 ? t : o.defaultBordered) !== null && n !== void 0 ? n : !0;
    }),
    mergedClsPrefixRef: F(() => (r == null ? void 0 : r.mergedClsPrefixRef.value) || Tl),
    namespaceRef: F(() => r == null ? void 0 : r.mergedNamespaceRef.value)
  };
}
function Il(e, o, r) {
  if (!o) {
    process.env.NODE_ENV !== "production" && Dr("use-style", "No style is specified.");
    return;
  }
  const t = Ho(), n = de(to, null), i = () => {
    const a = r == null ? void 0 : r.value;
    o.mount({
      id: a === void 0 ? e : a + e,
      head: !0,
      anchorMetaName: Ke,
      props: {
        bPrefix: a ? `.${a}-` : void 0
      },
      ssr: t
    }), n != null && n.preflightStyleDisabled || Xr.mount({
      id: "n-global",
      head: !0,
      anchorMetaName: Ke,
      ssr: t
    });
  };
  t ? i() : vr(i);
}
function et(e, o, r, t) {
  var n;
  r || Dr("useThemeClass", "cssVarsRef is not passed");
  const i = (n = de(to, null)) === null || n === void 0 ? void 0 : n.mergedThemeHashRef, a = K(""), l = Ho();
  let s;
  const c = `__${e}`, d = () => {
    let u = c;
    const g = o ? o.value : void 0, f = i == null ? void 0 : i.value;
    f && (u += "-" + f), g && (u += "-" + g);
    const { themeOverrides: p, builtinThemeOverrides: h } = t;
    p && (u += "-" + Co(JSON.stringify(p))), h && (u += "-" + Co(JSON.stringify(h))), a.value = u, s = () => {
      const x = r.value;
      let m = "";
      for (const y in x)
        m += `${y}: ${x[y]};`;
      j(`.${u}`, m).mount({
        id: u,
        ssr: l
      }), s = void 0;
    };
  };
  return $o(() => {
    d();
  }), {
    themeClass: a,
    onRender: () => {
      s == null || s();
    }
  };
}
const ot = te({
  name: "BaseIconSwitchTransition",
  setup(e, { slots: o }) {
    const r = Dn();
    return () => I(Sr, { name: "icon-switch-transition", appear: r.value }, o);
  }
}), {
  cubicBezierEaseInOut: zl
} = ro;
function Qe({
  originalTransform: e = "",
  left: o = 0,
  top: r = 0,
  transition: t = `all .3s ${zl} !important`
} = {}) {
  return [j("&.icon-switch-transition-enter-from, &.icon-switch-transition-leave-to", {
    transform: e + " scale(0.75)",
    left: o,
    top: r,
    opacity: 0
  }), j("&.icon-switch-transition-enter-to, &.icon-switch-transition-leave-from", {
    transform: `scale(1) ${e}`,
    left: o,
    top: r,
    opacity: 1
  }), j("&.icon-switch-transition-enter-active, &.icon-switch-transition-leave-active", {
    transformOrigin: "center",
    position: "absolute",
    left: o,
    top: r,
    transition: t
  })];
}
const kl = j([j("@keyframes loading-container-rotate", `
 to {
 -webkit-transform: rotate(360deg);
 transform: rotate(360deg);
 }
 `), j("@keyframes loading-layer-rotate", `
 12.5% {
 -webkit-transform: rotate(135deg);
 transform: rotate(135deg);
 }
 25% {
 -webkit-transform: rotate(270deg);
 transform: rotate(270deg);
 }
 37.5% {
 -webkit-transform: rotate(405deg);
 transform: rotate(405deg);
 }
 50% {
 -webkit-transform: rotate(540deg);
 transform: rotate(540deg);
 }
 62.5% {
 -webkit-transform: rotate(675deg);
 transform: rotate(675deg);
 }
 75% {
 -webkit-transform: rotate(810deg);
 transform: rotate(810deg);
 }
 87.5% {
 -webkit-transform: rotate(945deg);
 transform: rotate(945deg);
 }
 100% {
 -webkit-transform: rotate(1080deg);
 transform: rotate(1080deg);
 } 
 `), j("@keyframes loading-left-spin", `
 from {
 -webkit-transform: rotate(265deg);
 transform: rotate(265deg);
 }
 50% {
 -webkit-transform: rotate(130deg);
 transform: rotate(130deg);
 }
 to {
 -webkit-transform: rotate(265deg);
 transform: rotate(265deg);
 }
 `), j("@keyframes loading-right-spin", `
 from {
 -webkit-transform: rotate(-265deg);
 transform: rotate(-265deg);
 }
 50% {
 -webkit-transform: rotate(-130deg);
 transform: rotate(-130deg);
 }
 to {
 -webkit-transform: rotate(-265deg);
 transform: rotate(-265deg);
 }
 `), Ge("base-loading", `
 position: relative;
 line-height: 0;
 width: 1em;
 height: 1em;
 `, [D("transition-wrapper", `
 position: absolute;
 width: 100%;
 height: 100%;
 `, [Qe()]), D("container", `
 display: inline-flex;
 position: relative;
 direction: ltr;
 line-height: 0;
 animation: loading-container-rotate 1568.2352941176ms linear infinite;
 font-size: 0;
 letter-spacing: 0;
 white-space: nowrap;
 opacity: 1;
 width: 100%;
 height: 100%;
 `, [D("svg", `
 stroke: var(--n-text-color);
 fill: transparent;
 position: absolute;
 height: 100%;
 overflow: hidden;
 `), D("container-layer", `
 position: absolute;
 width: 100%;
 height: 100%;
 animation: loading-layer-rotate 5332ms cubic-bezier(0.4, 0, 0.2, 1) infinite both;
 `, [D("container-layer-left", `
 display: inline-flex;
 position: relative;
 width: 50%;
 height: 100%;
 overflow: hidden;
 `, [D("svg", `
 animation: loading-left-spin 1333ms cubic-bezier(0.4, 0, 0.2, 1) infinite both;
 width: 200%;
 `)]), D("container-layer-patch", `
 position: absolute;
 top: 0;
 left: 47.5%;
 box-sizing: border-box;
 width: 5%;
 height: 100%;
 overflow: hidden;
 `, [D("svg", `
 left: -900%;
 width: 2000%;
 transform: rotate(180deg);
 `)]), D("container-layer-right", `
 display: inline-flex;
 position: relative;
 width: 50%;
 height: 100%;
 overflow: hidden;
 `, [D("svg", `
 animation: loading-right-spin 1333ms cubic-bezier(0.4, 0, 0.2, 1) infinite both;
 left: -100%;
 width: 200%;
 `)])])]), D("placeholder", `
 position: absolute;
 left: 50%;
 top: 50%;
 transform: translateX(-50%) translateY(-50%);
 `, [Qe({
  left: "50%",
  top: "50%",
  originalTransform: "translateX(-50%) translateY(-50%)"
})])])]), _l = {
  strokeWidth: {
    type: Number,
    default: 28
  },
  stroke: {
    type: String,
    default: void 0
  }
}, Ml = te({
  name: "BaseLoading",
  props: Object.assign({ clsPrefix: {
    type: String,
    required: !0
  }, show: {
    type: Boolean,
    default: !0
  }, scale: {
    type: Number,
    default: 1
  }, radius: {
    type: Number,
    default: 100
  } }, _l),
  setup(e) {
    Il("-base-loading", kl, yr(e, "clsPrefix"));
  },
  render() {
    const { clsPrefix: e, radius: o, strokeWidth: r, stroke: t, scale: n } = this, i = o / n;
    return I(
      "div",
      { class: `${e}-base-loading`, role: "img", "aria-label": "loading" },
      I(ot, null, {
        default: () => this.show ? I(
          "div",
          { key: "icon", class: `${e}-base-loading__transition-wrapper` },
          I(
            "div",
            { class: `${e}-base-loading__container` },
            I(
              "div",
              { class: `${e}-base-loading__container-layer` },
              I(
                "div",
                { class: `${e}-base-loading__container-layer-left` },
                I(
                  "svg",
                  { class: `${e}-base-loading__svg`, viewBox: `0 0 ${2 * i} ${2 * i}`, xmlns: "http://www.w3.org/2000/svg", style: { color: t } },
                  I("circle", { fill: "none", stroke: "currentColor", "stroke-width": r, "stroke-linecap": "round", cx: i, cy: i, r: o - r / 2, "stroke-dasharray": 4.91 * o, "stroke-dashoffset": 2.46 * o })
                )
              ),
              I(
                "div",
                { class: `${e}-base-loading__container-layer-patch` },
                I(
                  "svg",
                  { class: `${e}-base-loading__svg`, viewBox: `0 0 ${2 * i} ${2 * i}`, xmlns: "http://www.w3.org/2000/svg", style: { color: t } },
                  I("circle", { fill: "none", stroke: "currentColor", "stroke-width": r, "stroke-linecap": "round", cx: i, cy: i, r: o - r / 2, "stroke-dasharray": 4.91 * o, "stroke-dashoffset": 2.46 * o })
                )
              ),
              I(
                "div",
                { class: `${e}-base-loading__container-layer-right` },
                I(
                  "svg",
                  { class: `${e}-base-loading__svg`, viewBox: `0 0 ${2 * i} ${2 * i}`, xmlns: "http://www.w3.org/2000/svg", style: { color: t } },
                  I("circle", { fill: "none", stroke: "currentColor", "stroke-width": r, "stroke-linecap": "round", cx: i, cy: i, r: o - r / 2, "stroke-dasharray": 4.91 * o, "stroke-dashoffset": 2.46 * o })
                )
              )
            )
          )
        ) : I("div", { key: "placeholder", class: `${e}-base-loading__placeholder` }, this.$slots)
      })
    );
  }
}), b = {
  neutralBase: "#000",
  neutralInvertBase: "#fff",
  neutralTextBase: "#fff",
  neutralPopover: "rgb(72, 72, 78)",
  neutralCard: "rgb(24, 24, 28)",
  neutralModal: "rgb(44, 44, 50)",
  neutralBody: "rgb(16, 16, 20)",
  alpha1: "0.9",
  alpha2: "0.82",
  alpha3: "0.52",
  alpha4: "0.38",
  alpha5: "0.28",
  alphaClose: "0.52",
  alphaDisabled: "0.38",
  alphaDisabledInput: "0.06",
  alphaPending: "0.09",
  alphaTablePending: "0.06",
  alphaTableStriped: "0.05",
  alphaPressed: "0.05",
  alphaAvatar: "0.18",
  alphaRail: "0.2",
  alphaProgressRail: "0.12",
  alphaBorder: "0.24",
  alphaDivider: "0.09",
  alphaInput: "0.1",
  alphaAction: "0.06",
  alphaTab: "0.04",
  alphaScrollbar: "0.2",
  alphaScrollbarHover: "0.3",
  alphaCode: "0.12",
  alphaTag: "0.2",
  // primary
  primaryHover: "#7fe7c4",
  primaryDefault: "#63e2b7",
  primaryActive: "#5acea7",
  primarySuppl: "rgb(42, 148, 125)",
  // info
  infoHover: "#8acbec",
  infoDefault: "#70c0e8",
  infoActive: "#66afd3",
  infoSuppl: "rgb(56, 137, 197)",
  // error
  errorHover: "#e98b8b",
  errorDefault: "#e88080",
  errorActive: "#e57272",
  errorSuppl: "rgb(208, 58, 82)",
  // warning
  warningHover: "#f5d599",
  warningDefault: "#f2c97d",
  warningActive: "#e6c260",
  warningSuppl: "rgb(240, 138, 0)",
  // success
  successHover: "#7fe7c4",
  successDefault: "#63e2b7",
  successActive: "#5acea7",
  successSuppl: "rgb(42, 148, 125)"
}, Ol = Z(b.neutralBase), rt = Z(b.neutralInvertBase), Bl = "rgba(" + rt.slice(0, 3).join(", ") + ", ";
function T(e) {
  return Bl + String(e) + ")";
}
function Al(e) {
  const o = Array.from(rt);
  return o[3] = Number(e), P(Ol, o);
}
const Fl = Object.assign(Object.assign({ name: "common" }, ro), {
  baseColor: b.neutralBase,
  // primary color
  primaryColor: b.primaryDefault,
  primaryColorHover: b.primaryHover,
  primaryColorPressed: b.primaryActive,
  primaryColorSuppl: b.primarySuppl,
  // info color
  infoColor: b.infoDefault,
  infoColorHover: b.infoHover,
  infoColorPressed: b.infoActive,
  infoColorSuppl: b.infoSuppl,
  // success color
  successColor: b.successDefault,
  successColorHover: b.successHover,
  successColorPressed: b.successActive,
  successColorSuppl: b.successSuppl,
  // warning color
  warningColor: b.warningDefault,
  warningColorHover: b.warningHover,
  warningColorPressed: b.warningActive,
  warningColorSuppl: b.warningSuppl,
  // error color
  errorColor: b.errorDefault,
  errorColorHover: b.errorHover,
  errorColorPressed: b.errorActive,
  errorColorSuppl: b.errorSuppl,
  // text color
  textColorBase: b.neutralTextBase,
  textColor1: T(b.alpha1),
  textColor2: T(b.alpha2),
  textColor3: T(b.alpha3),
  // textColor4: overlay(base.alpha4), // disabled, placeholder, icon
  // textColor5: overlay(base.alpha5),
  textColorDisabled: T(b.alpha4),
  placeholderColor: T(b.alpha4),
  placeholderColorDisabled: T(b.alpha5),
  iconColor: T(b.alpha4),
  iconColorDisabled: T(b.alpha5),
  iconColorHover: T(Number(b.alpha4) * 1.25),
  iconColorPressed: T(Number(b.alpha4) * 0.8),
  opacity1: b.alpha1,
  opacity2: b.alpha2,
  opacity3: b.alpha3,
  opacity4: b.alpha4,
  opacity5: b.alpha5,
  dividerColor: T(b.alphaDivider),
  borderColor: T(b.alphaBorder),
  // close
  closeIconColorHover: T(Number(b.alphaClose)),
  closeIconColor: T(Number(b.alphaClose)),
  closeIconColorPressed: T(Number(b.alphaClose)),
  closeColorHover: "rgba(255, 255, 255, .12)",
  closeColorPressed: "rgba(255, 255, 255, .08)",
  // clear
  clearColor: T(b.alpha4),
  clearColorHover: A(T(b.alpha4), { alpha: 1.25 }),
  clearColorPressed: A(T(b.alpha4), { alpha: 0.8 }),
  scrollbarColor: T(b.alphaScrollbar),
  scrollbarColorHover: T(b.alphaScrollbarHover),
  scrollbarWidth: "5px",
  scrollbarHeight: "5px",
  scrollbarBorderRadius: "5px",
  progressRailColor: T(b.alphaProgressRail),
  railColor: T(b.alphaRail),
  popoverColor: b.neutralPopover,
  tableColor: b.neutralCard,
  cardColor: b.neutralCard,
  modalColor: b.neutralModal,
  bodyColor: b.neutralBody,
  tagColor: Al(b.alphaTag),
  avatarColor: T(b.alphaAvatar),
  invertedColor: b.neutralBase,
  inputColor: T(b.alphaInput),
  codeColor: T(b.alphaCode),
  tabColor: T(b.alphaTab),
  actionColor: T(b.alphaAction),
  tableHeaderColor: T(b.alphaAction),
  hoverColor: T(b.alphaPending),
  tableColorHover: T(b.alphaTablePending),
  tableColorStriped: T(b.alphaTableStriped),
  pressedColor: T(b.alphaPressed),
  opacityDisabled: b.alphaDisabled,
  inputColorDisabled: T(b.alphaDisabledInput),
  buttonColor2: "rgba(255, 255, 255, .08)",
  buttonColor2Hover: "rgba(255, 255, 255, .12)",
  buttonColor2Pressed: "rgba(255, 255, 255, .08)",
  boxShadow1: "0 1px 2px -2px rgba(0, 0, 0, .24), 0 3px 6px 0 rgba(0, 0, 0, .18), 0 5px 12px 4px rgba(0, 0, 0, .12)",
  boxShadow2: "0 3px 6px -4px rgba(0, 0, 0, .24), 0 6px 12px 0 rgba(0, 0, 0, .16), 0 9px 18px 8px rgba(0, 0, 0, .10)",
  boxShadow3: "0 6px 16px -9px rgba(0, 0, 0, .08), 0 9px 28px 0 rgba(0, 0, 0, .05), 0 12px 48px 16px rgba(0, 0, 0, .03)"
}), C = Fl, $ = {
  neutralBase: "#FFF",
  neutralInvertBase: "#000",
  neutralTextBase: "#000",
  neutralPopover: "#fff",
  neutralCard: "#fff",
  neutralModal: "#fff",
  neutralBody: "#fff",
  alpha1: "0.82",
  alpha2: "0.72",
  alpha3: "0.38",
  alpha4: "0.24",
  alpha5: "0.18",
  alphaClose: "0.6",
  alphaDisabled: "0.5",
  alphaDisabledInput: "0.02",
  alphaPending: "0.05",
  alphaTablePending: "0.02",
  alphaPressed: "0.07",
  alphaAvatar: "0.2",
  alphaRail: "0.14",
  alphaProgressRail: ".08",
  alphaBorder: "0.12",
  alphaDivider: "0.06",
  alphaInput: "0",
  alphaAction: "0.02",
  alphaTab: "0.04",
  alphaScrollbar: "0.25",
  alphaScrollbarHover: "0.4",
  alphaCode: "0.05",
  alphaTag: "0.02",
  // primary
  primaryHover: "#36ad6a",
  primaryDefault: "#18a058",
  primaryActive: "#0c7a43",
  primarySuppl: "#36ad6a",
  // info
  infoHover: "#4098fc",
  infoDefault: "#2080f0",
  infoActive: "#1060c9",
  infoSuppl: "#4098fc",
  // error
  errorHover: "#de576d",
  errorDefault: "#d03050",
  errorActive: "#ab1f3f",
  errorSuppl: "#de576d",
  // warning
  warningHover: "#fcb040",
  warningDefault: "#f0a020",
  warningActive: "#c97c10",
  warningSuppl: "#fcb040",
  // success
  successHover: "#36ad6a",
  successDefault: "#18a058",
  successActive: "#0c7a43",
  successSuppl: "#36ad6a"
}, El = Z($.neutralBase), tt = Z($.neutralInvertBase), Wl = "rgba(" + tt.slice(0, 3).join(", ") + ", ";
function or(e) {
  return Wl + String(e) + ")";
}
function L(e) {
  const o = Array.from(tt);
  return o[3] = Number(e), P(El, o);
}
const Ll = Object.assign(Object.assign({ name: "common" }, ro), {
  baseColor: $.neutralBase,
  // primary color
  primaryColor: $.primaryDefault,
  primaryColorHover: $.primaryHover,
  primaryColorPressed: $.primaryActive,
  primaryColorSuppl: $.primarySuppl,
  // info color
  infoColor: $.infoDefault,
  infoColorHover: $.infoHover,
  infoColorPressed: $.infoActive,
  infoColorSuppl: $.infoSuppl,
  // success color
  successColor: $.successDefault,
  successColorHover: $.successHover,
  successColorPressed: $.successActive,
  successColorSuppl: $.successSuppl,
  // warning color
  warningColor: $.warningDefault,
  warningColorHover: $.warningHover,
  warningColorPressed: $.warningActive,
  warningColorSuppl: $.warningSuppl,
  // error color
  errorColor: $.errorDefault,
  errorColorHover: $.errorHover,
  errorColorPressed: $.errorActive,
  errorColorSuppl: $.errorSuppl,
  // text color
  textColorBase: $.neutralTextBase,
  textColor1: "rgb(31, 34, 37)",
  textColor2: "rgb(51, 54, 57)",
  textColor3: "rgb(118, 124, 130)",
  // textColor4: neutral(base.alpha4), // disabled, placeholder, icon
  // textColor5: neutral(base.alpha5),
  textColorDisabled: L($.alpha4),
  placeholderColor: L($.alpha4),
  placeholderColorDisabled: L($.alpha5),
  iconColor: L($.alpha4),
  iconColorHover: A(L($.alpha4), { lightness: 0.75 }),
  iconColorPressed: A(L($.alpha4), { lightness: 0.9 }),
  iconColorDisabled: L($.alpha5),
  opacity1: $.alpha1,
  opacity2: $.alpha2,
  opacity3: $.alpha3,
  opacity4: $.alpha4,
  opacity5: $.alpha5,
  dividerColor: "rgb(239, 239, 245)",
  borderColor: "rgb(224, 224, 230)",
  // close
  closeIconColor: L(Number($.alphaClose)),
  closeIconColorHover: L(Number($.alphaClose)),
  closeIconColorPressed: L(Number($.alphaClose)),
  closeColorHover: "rgba(0, 0, 0, .09)",
  closeColorPressed: "rgba(0, 0, 0, .13)",
  // clear
  clearColor: L($.alpha4),
  clearColorHover: A(L($.alpha4), { lightness: 0.75 }),
  clearColorPressed: A(L($.alpha4), { lightness: 0.9 }),
  scrollbarColor: or($.alphaScrollbar),
  scrollbarColorHover: or($.alphaScrollbarHover),
  scrollbarWidth: "5px",
  scrollbarHeight: "5px",
  scrollbarBorderRadius: "5px",
  progressRailColor: L($.alphaProgressRail),
  railColor: "rgb(219, 219, 223)",
  popoverColor: $.neutralPopover,
  tableColor: $.neutralCard,
  cardColor: $.neutralCard,
  modalColor: $.neutralModal,
  bodyColor: $.neutralBody,
  tagColor: "#eee",
  avatarColor: L($.alphaAvatar),
  invertedColor: "rgb(0, 20, 40)",
  inputColor: L($.alphaInput),
  codeColor: "rgb(244, 244, 248)",
  tabColor: "rgb(247, 247, 250)",
  actionColor: "rgb(250, 250, 252)",
  tableHeaderColor: "rgb(250, 250, 252)",
  hoverColor: "rgb(243, 243, 245)",
  // use color with alpha since it can be nested with header filter & sorter effect
  tableColorHover: "rgba(0, 0, 100, 0.03)",
  tableColorStriped: "rgba(0, 0, 100, 0.02)",
  pressedColor: "rgb(237, 237, 239)",
  opacityDisabled: $.alphaDisabled,
  inputColorDisabled: "rgb(250, 250, 252)",
  // secondary button color
  // can also be used in tertiary button & quaternary button
  buttonColor2: "rgba(46, 51, 56, .05)",
  buttonColor2Hover: "rgba(46, 51, 56, .09)",
  buttonColor2Pressed: "rgba(46, 51, 56, .13)",
  boxShadow1: "0 1px 2px -2px rgba(0, 0, 0, .08), 0 3px 6px 0 rgba(0, 0, 0, .06), 0 5px 12px 4px rgba(0, 0, 0, .04)",
  boxShadow2: "0 3px 6px -4px rgba(0, 0, 0, .12), 0 6px 16px 0 rgba(0, 0, 0, .08), 0 9px 28px 8px rgba(0, 0, 0, .05)",
  boxShadow3: "0 6px 16px -9px rgba(0, 0, 0, .08), 0 9px 28px 0 rgba(0, 0, 0, .05), 0 12px 48px 16px rgba(0, 0, 0, .03)"
}), ko = Ll, Rl = {
  iconSizeSmall: "34px",
  iconSizeMedium: "40px",
  iconSizeLarge: "46px",
  iconSizeHuge: "52px"
}, nt = (e) => {
  const { textColorDisabled: o, iconColor: r, textColor2: t, fontSizeSmall: n, fontSizeMedium: i, fontSizeLarge: a, fontSizeHuge: l } = e;
  return Object.assign(Object.assign({}, Rl), {
    fontSizeSmall: n,
    fontSizeMedium: i,
    fontSizeLarge: a,
    fontSizeHuge: l,
    textColor: o,
    iconColor: r,
    extraTextColor: t
  });
}, jl = {
  name: "Empty",
  common: ko,
  self: nt
}, Vl = jl, Nl = {
  name: "Empty",
  common: C,
  self: nt
}, ue = Nl, Gl = (e) => {
  const { scrollbarColor: o, scrollbarColorHover: r } = e;
  return {
    color: o,
    colorHover: r
  };
}, Ul = {
  name: "Scrollbar",
  common: C,
  self: Gl
}, G = Ul, ql = {
  height: "calc(var(--n-option-height) * 7.6)",
  paddingSmall: "4px 0",
  paddingMedium: "4px 0",
  paddingLarge: "4px 0",
  paddingHuge: "4px 0",
  optionPaddingSmall: "0 12px",
  optionPaddingMedium: "0 12px",
  optionPaddingLarge: "0 12px",
  optionPaddingHuge: "0 12px",
  loadingSize: "18px"
}, Kl = (e) => {
  const { borderRadius: o, popoverColor: r, textColor3: t, dividerColor: n, textColor2: i, primaryColorPressed: a, textColorDisabled: l, primaryColor: s, opacityDisabled: c, hoverColor: d, fontSizeSmall: u, fontSizeMedium: g, fontSizeLarge: f, fontSizeHuge: p, heightSmall: h, heightMedium: x, heightLarge: m, heightHuge: y } = e;
  return Object.assign(Object.assign({}, ql), { optionFontSizeSmall: u, optionFontSizeMedium: g, optionFontSizeLarge: f, optionFontSizeHuge: p, optionHeightSmall: h, optionHeightMedium: x, optionHeightLarge: m, optionHeightHuge: y, borderRadius: o, color: r, groupHeaderTextColor: t, actionDividerColor: n, optionTextColor: i, optionTextColorPressed: a, optionTextColorDisabled: l, optionTextColorActive: s, optionOpacityDisabled: c, optionCheckColor: s, optionColorPending: d, optionColorActive: "rgba(0, 0, 0, 0)", optionColorActivePending: d, actionTextColor: i, loadingColor: s });
}, Ql = {
  name: "InternalSelectMenu",
  common: C,
  peers: {
    Scrollbar: G,
    Empty: ue
  },
  self: Kl
}, Fe = Ql, Jl = {
  space: "6px",
  spaceArrow: "10px",
  arrowOffset: "10px",
  arrowOffsetVertical: "10px",
  arrowHeight: "6px",
  padding: "8px 14px"
}, Yl = (e) => {
  const { boxShadow2: o, popoverColor: r, textColor2: t, borderRadius: n, fontSize: i, dividerColor: a } = e;
  return Object.assign(Object.assign({}, Jl), {
    fontSize: i,
    borderRadius: n,
    color: r,
    dividerColor: a,
    textColor: t,
    boxShadow: o
  });
}, Xl = {
  name: "Popover",
  common: C,
  self: Yl
}, fe = Xl, Zl = {
  closeIconSizeTiny: "12px",
  closeIconSizeSmall: "12px",
  closeIconSizeMedium: "14px",
  closeIconSizeLarge: "14px",
  closeSizeTiny: "16px",
  closeSizeSmall: "16px",
  closeSizeMedium: "18px",
  closeSizeLarge: "18px",
  padding: "0 7px",
  closeMargin: "0 0 0 4px",
  closeMarginRtl: "0 4px 0 0"
}, es = {
  name: "Tag",
  common: C,
  self(e) {
    const { textColor2: o, primaryColorHover: r, primaryColorPressed: t, primaryColor: n, infoColor: i, successColor: a, warningColor: l, errorColor: s, baseColor: c, borderColor: d, tagColor: u, opacityDisabled: g, closeIconColor: f, closeIconColorHover: p, closeIconColorPressed: h, closeColorHover: x, closeColorPressed: m, borderRadiusSmall: y, fontSizeMini: w, fontSizeTiny: z, fontSizeSmall: _, fontSizeMedium: B, heightMini: k, heightTiny: v, heightSmall: H, heightMedium: M, buttonColor2Hover: E, buttonColor2Pressed: W, fontWeightStrong: q } = e;
    return Object.assign(Object.assign({}, Zl), {
      closeBorderRadius: y,
      heightTiny: k,
      heightSmall: v,
      heightMedium: H,
      heightLarge: M,
      borderRadius: y,
      opacityDisabled: g,
      fontSizeTiny: w,
      fontSizeSmall: z,
      fontSizeMedium: _,
      fontSizeLarge: B,
      fontWeightStrong: q,
      // checked
      textColorCheckable: o,
      textColorHoverCheckable: o,
      textColorPressedCheckable: o,
      textColorChecked: c,
      colorCheckable: "#0000",
      colorHoverCheckable: E,
      colorPressedCheckable: W,
      colorChecked: n,
      colorCheckedHover: r,
      colorCheckedPressed: t,
      // default
      border: `1px solid ${d}`,
      textColor: o,
      color: u,
      colorBordered: "#0000",
      closeIconColor: f,
      closeIconColorHover: p,
      closeIconColorPressed: h,
      closeColorHover: x,
      closeColorPressed: m,
      borderPrimary: `1px solid ${S(n, { alpha: 0.3 })}`,
      textColorPrimary: n,
      colorPrimary: S(n, { alpha: 0.16 }),
      colorBorderedPrimary: "#0000",
      closeIconColorPrimary: A(n, { lightness: 0.7 }),
      closeIconColorHoverPrimary: A(n, { lightness: 0.7 }),
      closeIconColorPressedPrimary: A(n, {
        lightness: 0.7
      }),
      closeColorHoverPrimary: S(n, { alpha: 0.16 }),
      closeColorPressedPrimary: S(n, { alpha: 0.12 }),
      borderInfo: `1px solid ${S(i, { alpha: 0.3 })}`,
      textColorInfo: i,
      colorInfo: S(i, { alpha: 0.16 }),
      colorBorderedInfo: "#0000",
      closeIconColorInfo: A(i, { alpha: 0.7 }),
      closeIconColorHoverInfo: A(i, { alpha: 0.7 }),
      closeIconColorPressedInfo: A(i, { alpha: 0.7 }),
      closeColorHoverInfo: S(i, { alpha: 0.16 }),
      closeColorPressedInfo: S(i, { alpha: 0.12 }),
      borderSuccess: `1px solid ${S(a, { alpha: 0.3 })}`,
      textColorSuccess: a,
      colorSuccess: S(a, { alpha: 0.16 }),
      colorBorderedSuccess: "#0000",
      closeIconColorSuccess: A(a, { alpha: 0.7 }),
      closeIconColorHoverSuccess: A(a, { alpha: 0.7 }),
      closeIconColorPressedSuccess: A(a, { alpha: 0.7 }),
      closeColorHoverSuccess: S(a, { alpha: 0.16 }),
      closeColorPressedSuccess: S(a, { alpha: 0.12 }),
      borderWarning: `1px solid ${S(l, { alpha: 0.3 })}`,
      textColorWarning: l,
      colorWarning: S(l, { alpha: 0.16 }),
      colorBorderedWarning: "#0000",
      closeIconColorWarning: A(l, { alpha: 0.7 }),
      closeIconColorHoverWarning: A(l, { alpha: 0.7 }),
      closeIconColorPressedWarning: A(l, { alpha: 0.7 }),
      closeColorHoverWarning: S(l, { alpha: 0.16 }),
      closeColorPressedWarning: S(l, { alpha: 0.11 }),
      borderError: `1px solid ${S(s, { alpha: 0.3 })}`,
      textColorError: s,
      colorError: S(s, { alpha: 0.16 }),
      colorBorderedError: "#0000",
      closeIconColorError: A(s, { alpha: 0.7 }),
      closeIconColorHoverError: A(s, { alpha: 0.7 }),
      closeIconColorPressedError: A(s, { alpha: 0.7 }),
      closeColorHoverError: S(s, { alpha: 0.16 }),
      closeColorPressedError: S(s, { alpha: 0.12 })
    });
  }
}, it = es, os = {
  paddingSingle: "0 26px 0 12px",
  paddingMultiple: "3px 26px 0 12px",
  clearSize: "16px",
  arrowSize: "16px"
}, rs = {
  name: "InternalSelection",
  common: C,
  peers: {
    Popover: fe
  },
  self(e) {
    const { borderRadius: o, textColor2: r, textColorDisabled: t, inputColor: n, inputColorDisabled: i, primaryColor: a, primaryColorHover: l, warningColor: s, warningColorHover: c, errorColor: d, errorColorHover: u, iconColor: g, iconColorDisabled: f, clearColor: p, clearColorHover: h, clearColorPressed: x, placeholderColor: m, placeholderColorDisabled: y, fontSizeTiny: w, fontSizeSmall: z, fontSizeMedium: _, fontSizeLarge: B, heightTiny: k, heightSmall: v, heightMedium: H, heightLarge: M } = e;
    return Object.assign(Object.assign({}, os), {
      fontSizeTiny: w,
      fontSizeSmall: z,
      fontSizeMedium: _,
      fontSizeLarge: B,
      heightTiny: k,
      heightSmall: v,
      heightMedium: H,
      heightLarge: M,
      borderRadius: o,
      // default
      textColor: r,
      textColorDisabled: t,
      placeholderColor: m,
      placeholderColorDisabled: y,
      color: n,
      colorDisabled: i,
      colorActive: S(a, { alpha: 0.1 }),
      border: "1px solid #0000",
      borderHover: `1px solid ${l}`,
      borderActive: `1px solid ${a}`,
      borderFocus: `1px solid ${l}`,
      boxShadowHover: "none",
      boxShadowActive: `0 0 8px 0 ${S(a, {
        alpha: 0.4
      })}`,
      boxShadowFocus: `0 0 8px 0 ${S(a, {
        alpha: 0.4
      })}`,
      caretColor: a,
      arrowColor: g,
      arrowColorDisabled: f,
      loadingColor: a,
      // warning
      borderWarning: `1px solid ${s}`,
      borderHoverWarning: `1px solid ${c}`,
      borderActiveWarning: `1px solid ${s}`,
      borderFocusWarning: `1px solid ${c}`,
      boxShadowHoverWarning: "none",
      boxShadowActiveWarning: `0 0 8px 0 ${S(s, {
        alpha: 0.4
      })}`,
      boxShadowFocusWarning: `0 0 8px 0 ${S(s, {
        alpha: 0.4
      })}`,
      colorActiveWarning: S(s, { alpha: 0.1 }),
      caretColorWarning: s,
      // error
      borderError: `1px solid ${d}`,
      borderHoverError: `1px solid ${u}`,
      borderActiveError: `1px solid ${d}`,
      borderFocusError: `1px solid ${u}`,
      boxShadowHoverError: "none",
      boxShadowActiveError: `0 0 8px 0 ${S(d, {
        alpha: 0.4
      })}`,
      boxShadowFocusError: `0 0 8px 0 ${S(d, {
        alpha: 0.4
      })}`,
      colorActiveError: S(d, { alpha: 0.1 }),
      caretColorError: d,
      clearColor: p,
      clearColorHover: h,
      clearColorPressed: x
    });
  }
}, _o = rs, ts = {
  iconMargin: "11px 8px 0 12px",
  iconMarginRtl: "11px 12px 0 8px",
  iconSize: "24px",
  closeIconSize: "16px",
  closeSize: "20px",
  closeMargin: "13px 14px 0 0",
  closeMarginRtl: "13px 0 0 14px",
  padding: "13px"
}, ns = {
  name: "Alert",
  common: C,
  self(e) {
    const { lineHeight: o, borderRadius: r, fontWeightStrong: t, dividerColor: n, inputColor: i, textColor1: a, textColor2: l, closeColorHover: s, closeColorPressed: c, closeIconColor: d, closeIconColorHover: u, closeIconColorPressed: g, infoColorSuppl: f, successColorSuppl: p, warningColorSuppl: h, errorColorSuppl: x, fontSize: m } = e;
    return Object.assign(Object.assign({}, ts), {
      fontSize: m,
      lineHeight: o,
      titleFontWeight: t,
      borderRadius: r,
      border: `1px solid ${n}`,
      color: i,
      titleTextColor: a,
      iconColor: l,
      contentTextColor: l,
      closeBorderRadius: r,
      closeColorHover: s,
      closeColorPressed: c,
      closeIconColor: d,
      closeIconColorHover: u,
      closeIconColorPressed: g,
      borderInfo: `1px solid ${S(f, { alpha: 0.35 })}`,
      colorInfo: S(f, { alpha: 0.25 }),
      titleTextColorInfo: a,
      iconColorInfo: f,
      contentTextColorInfo: l,
      closeColorHoverInfo: s,
      closeColorPressedInfo: c,
      closeIconColorInfo: d,
      closeIconColorHoverInfo: u,
      closeIconColorPressedInfo: g,
      borderSuccess: `1px solid ${S(p, {
        alpha: 0.35
      })}`,
      colorSuccess: S(p, { alpha: 0.25 }),
      titleTextColorSuccess: a,
      iconColorSuccess: p,
      contentTextColorSuccess: l,
      closeColorHoverSuccess: s,
      closeColorPressedSuccess: c,
      closeIconColorSuccess: d,
      closeIconColorHoverSuccess: u,
      closeIconColorPressedSuccess: g,
      borderWarning: `1px solid ${S(h, {
        alpha: 0.35
      })}`,
      colorWarning: S(h, { alpha: 0.25 }),
      titleTextColorWarning: a,
      iconColorWarning: h,
      contentTextColorWarning: l,
      closeColorHoverWarning: s,
      closeColorPressedWarning: c,
      closeIconColorWarning: d,
      closeIconColorHoverWarning: u,
      closeIconColorPressedWarning: g,
      borderError: `1px solid ${S(x, { alpha: 0.35 })}`,
      colorError: S(x, { alpha: 0.25 }),
      titleTextColorError: a,
      iconColorError: x,
      contentTextColorError: l,
      closeColorHoverError: s,
      closeColorPressedError: c,
      closeIconColorError: d,
      closeIconColorHoverError: u,
      closeIconColorPressedError: g
    });
  }
}, is = ns, as = {
  linkFontSize: "13px",
  linkPadding: "0 0 0 16px",
  railWidth: "4px"
}, ls = (e) => {
  const { borderRadius: o, railColor: r, primaryColor: t, primaryColorHover: n, primaryColorPressed: i, textColor2: a } = e;
  return Object.assign(Object.assign({}, as), {
    borderRadius: o,
    railColor: r,
    railColorActive: t,
    linkColor: S(t, { alpha: 0.15 }),
    linkTextColor: a,
    linkTextColorHover: n,
    linkTextColorPressed: i,
    linkTextColorActive: t
  });
}, ss = {
  name: "Anchor",
  common: C,
  self: ls
}, cs = ss, ds = {
  paddingTiny: "0 8px",
  paddingSmall: "0 10px",
  paddingMedium: "0 12px",
  paddingLarge: "0 14px",
  clearSize: "16px"
}, ps = {
  name: "Input",
  common: C,
  self(e) {
    const { textColor2: o, textColor3: r, textColorDisabled: t, primaryColor: n, primaryColorHover: i, inputColor: a, inputColorDisabled: l, warningColor: s, warningColorHover: c, errorColor: d, errorColorHover: u, borderRadius: g, lineHeight: f, fontSizeTiny: p, fontSizeSmall: h, fontSizeMedium: x, fontSizeLarge: m, heightTiny: y, heightSmall: w, heightMedium: z, heightLarge: _, clearColor: B, clearColorHover: k, clearColorPressed: v, placeholderColor: H, placeholderColorDisabled: M, iconColor: E, iconColorDisabled: W, iconColorHover: q, iconColorPressed: Pe } = e;
    return Object.assign(Object.assign({}, ds), {
      countTextColorDisabled: t,
      countTextColor: r,
      heightTiny: y,
      heightSmall: w,
      heightMedium: z,
      heightLarge: _,
      fontSizeTiny: p,
      fontSizeSmall: h,
      fontSizeMedium: x,
      fontSizeLarge: m,
      lineHeight: f,
      lineHeightTextarea: f,
      borderRadius: g,
      iconSize: "16px",
      groupLabelColor: a,
      textColor: o,
      textColorDisabled: t,
      textDecorationColor: o,
      groupLabelTextColor: o,
      caretColor: n,
      placeholderColor: H,
      placeholderColorDisabled: M,
      color: a,
      colorDisabled: l,
      colorFocus: S(n, { alpha: 0.1 }),
      groupLabelBorder: "1px solid #0000",
      border: "1px solid #0000",
      borderHover: `1px solid ${i}`,
      borderDisabled: "1px solid #0000",
      borderFocus: `1px solid ${i}`,
      boxShadowFocus: `0 0 8px 0 ${S(n, { alpha: 0.3 })}`,
      loadingColor: n,
      // warning
      loadingColorWarning: s,
      borderWarning: `1px solid ${s}`,
      borderHoverWarning: `1px solid ${c}`,
      colorFocusWarning: S(s, { alpha: 0.1 }),
      borderFocusWarning: `1px solid ${c}`,
      boxShadowFocusWarning: `0 0 8px 0 ${S(s, {
        alpha: 0.3
      })}`,
      caretColorWarning: s,
      // error
      loadingColorError: d,
      borderError: `1px solid ${d}`,
      borderHoverError: `1px solid ${u}`,
      colorFocusError: S(d, { alpha: 0.1 }),
      borderFocusError: `1px solid ${u}`,
      boxShadowFocusError: `0 0 8px 0 ${S(d, {
        alpha: 0.3
      })}`,
      caretColorError: d,
      clearColor: B,
      clearColorHover: k,
      clearColorPressed: v,
      iconColor: E,
      iconColorDisabled: W,
      iconColorHover: q,
      iconColorPressed: Pe,
      suffixTextColor: o
    });
  }
}, Q = ps;
function us(e) {
  const { boxShadow2: o } = e;
  return {
    menuBoxShadow: o
  };
}
const fs = {
  name: "AutoComplete",
  common: C,
  peers: {
    InternalSelectMenu: Fe,
    Input: Q
  },
  self: us
}, gs = fs, ms = (e) => {
  const { borderRadius: o, avatarColor: r, cardColor: t, fontSize: n, heightTiny: i, heightSmall: a, heightMedium: l, heightLarge: s, heightHuge: c, modalColor: d, popoverColor: u } = e;
  return {
    borderRadius: o,
    fontSize: n,
    border: `2px solid ${t}`,
    heightTiny: i,
    heightSmall: a,
    heightMedium: l,
    heightLarge: s,
    heightHuge: c,
    color: P(t, r),
    colorModal: P(d, r),
    colorPopover: P(u, r)
  };
}, Cs = {
  name: "Avatar",
  common: C,
  self: ms
}, at = Cs, hs = () => ({
  gap: "-12px"
}), xs = {
  name: "AvatarGroup",
  common: C,
  peers: {
    Avatar: at
  },
  self: hs
}, bs = xs, vs = {
  width: "44px",
  height: "44px",
  borderRadius: "22px",
  iconSize: "26px"
}, Ss = {
  name: "BackTop",
  common: C,
  self(e) {
    const { popoverColor: o, textColor2: r, primaryColorHover: t, primaryColorPressed: n } = e;
    return Object.assign(Object.assign({}, vs), { color: o, textColor: r, iconColor: r, iconColorHover: t, iconColorPressed: n, boxShadow: "0 2px 8px 0px rgba(0, 0, 0, .12)", boxShadowHover: "0 2px 12px 0px rgba(0, 0, 0, .18)", boxShadowPressed: "0 2px 12px 0px rgba(0, 0, 0, .18)" });
  }
}, ys = Ss, $s = {
  name: "Badge",
  common: C,
  self(e) {
    const { errorColorSuppl: o, infoColorSuppl: r, successColorSuppl: t, warningColorSuppl: n, fontFamily: i } = e;
    return {
      color: o,
      colorInfo: r,
      colorSuccess: t,
      colorError: o,
      colorWarning: n,
      fontSize: "12px",
      fontFamily: i
    };
  }
}, Ps = $s, ws = {
  fontWeightActive: "400"
}, Hs = (e) => {
  const { fontSize: o, textColor3: r, textColor2: t, borderRadius: n, buttonColor2Hover: i, buttonColor2Pressed: a } = e;
  return Object.assign(Object.assign({}, ws), { fontSize: o, itemLineHeight: "1.25", itemTextColor: r, itemTextColorHover: t, itemTextColorPressed: t, itemTextColorActive: t, itemBorderRadius: n, itemColorHover: i, itemColorPressed: a, separatorColor: r });
}, Ds = {
  name: "Breadcrumb",
  common: C,
  self: Hs
}, Ts = Ds, Is = {
  paddingTiny: "0 6px",
  paddingSmall: "0 10px",
  paddingMedium: "0 14px",
  paddingLarge: "0 18px",
  paddingRoundTiny: "0 10px",
  paddingRoundSmall: "0 14px",
  paddingRoundMedium: "0 18px",
  paddingRoundLarge: "0 22px",
  iconMarginTiny: "6px",
  iconMarginSmall: "6px",
  iconMarginMedium: "6px",
  iconMarginLarge: "6px",
  iconSizeTiny: "14px",
  iconSizeSmall: "18px",
  iconSizeMedium: "18px",
  iconSizeLarge: "20px",
  rippleDuration: ".6s"
}, zs = (e) => {
  const { heightTiny: o, heightSmall: r, heightMedium: t, heightLarge: n, borderRadius: i, fontSizeTiny: a, fontSizeSmall: l, fontSizeMedium: s, fontSizeLarge: c, opacityDisabled: d, textColor2: u, textColor3: g, primaryColorHover: f, primaryColorPressed: p, borderColor: h, primaryColor: x, baseColor: m, infoColor: y, infoColorHover: w, infoColorPressed: z, successColor: _, successColorHover: B, successColorPressed: k, warningColor: v, warningColorHover: H, warningColorPressed: M, errorColor: E, errorColorHover: W, errorColorPressed: q, fontWeight: Pe, buttonColor2: Ee, buttonColor2Hover: we, buttonColor2Pressed: He, fontWeightStrong: Y } = e;
  return Object.assign(Object.assign({}, Is), {
    heightTiny: o,
    heightSmall: r,
    heightMedium: t,
    heightLarge: n,
    borderRadiusTiny: i,
    borderRadiusSmall: i,
    borderRadiusMedium: i,
    borderRadiusLarge: i,
    fontSizeTiny: a,
    fontSizeSmall: l,
    fontSizeMedium: s,
    fontSizeLarge: c,
    opacityDisabled: d,
    // secondary
    colorOpacitySecondary: "0.16",
    colorOpacitySecondaryHover: "0.22",
    colorOpacitySecondaryPressed: "0.28",
    colorSecondary: Ee,
    colorSecondaryHover: we,
    colorSecondaryPressed: He,
    // tertiary
    colorTertiary: Ee,
    colorTertiaryHover: we,
    colorTertiaryPressed: He,
    // quaternary
    colorQuaternary: "#0000",
    colorQuaternaryHover: we,
    colorQuaternaryPressed: He,
    // default type
    color: "#0000",
    colorHover: "#0000",
    colorPressed: "#0000",
    colorFocus: "#0000",
    colorDisabled: "#0000",
    textColor: u,
    textColorTertiary: g,
    textColorHover: f,
    textColorPressed: p,
    textColorFocus: f,
    textColorDisabled: u,
    textColorText: u,
    textColorTextHover: f,
    textColorTextPressed: p,
    textColorTextFocus: f,
    textColorTextDisabled: u,
    textColorGhost: u,
    textColorGhostHover: f,
    textColorGhostPressed: p,
    textColorGhostFocus: f,
    textColorGhostDisabled: u,
    border: `1px solid ${h}`,
    borderHover: `1px solid ${f}`,
    borderPressed: `1px solid ${p}`,
    borderFocus: `1px solid ${f}`,
    borderDisabled: `1px solid ${h}`,
    rippleColor: x,
    // primary
    colorPrimary: x,
    colorHoverPrimary: f,
    colorPressedPrimary: p,
    colorFocusPrimary: f,
    colorDisabledPrimary: x,
    textColorPrimary: m,
    textColorHoverPrimary: m,
    textColorPressedPrimary: m,
    textColorFocusPrimary: m,
    textColorDisabledPrimary: m,
    textColorTextPrimary: x,
    textColorTextHoverPrimary: f,
    textColorTextPressedPrimary: p,
    textColorTextFocusPrimary: f,
    textColorTextDisabledPrimary: u,
    textColorGhostPrimary: x,
    textColorGhostHoverPrimary: f,
    textColorGhostPressedPrimary: p,
    textColorGhostFocusPrimary: f,
    textColorGhostDisabledPrimary: x,
    borderPrimary: `1px solid ${x}`,
    borderHoverPrimary: `1px solid ${f}`,
    borderPressedPrimary: `1px solid ${p}`,
    borderFocusPrimary: `1px solid ${f}`,
    borderDisabledPrimary: `1px solid ${x}`,
    rippleColorPrimary: x,
    // info
    colorInfo: y,
    colorHoverInfo: w,
    colorPressedInfo: z,
    colorFocusInfo: w,
    colorDisabledInfo: y,
    textColorInfo: m,
    textColorHoverInfo: m,
    textColorPressedInfo: m,
    textColorFocusInfo: m,
    textColorDisabledInfo: m,
    textColorTextInfo: y,
    textColorTextHoverInfo: w,
    textColorTextPressedInfo: z,
    textColorTextFocusInfo: w,
    textColorTextDisabledInfo: u,
    textColorGhostInfo: y,
    textColorGhostHoverInfo: w,
    textColorGhostPressedInfo: z,
    textColorGhostFocusInfo: w,
    textColorGhostDisabledInfo: y,
    borderInfo: `1px solid ${y}`,
    borderHoverInfo: `1px solid ${w}`,
    borderPressedInfo: `1px solid ${z}`,
    borderFocusInfo: `1px solid ${w}`,
    borderDisabledInfo: `1px solid ${y}`,
    rippleColorInfo: y,
    // success
    colorSuccess: _,
    colorHoverSuccess: B,
    colorPressedSuccess: k,
    colorFocusSuccess: B,
    colorDisabledSuccess: _,
    textColorSuccess: m,
    textColorHoverSuccess: m,
    textColorPressedSuccess: m,
    textColorFocusSuccess: m,
    textColorDisabledSuccess: m,
    textColorTextSuccess: _,
    textColorTextHoverSuccess: B,
    textColorTextPressedSuccess: k,
    textColorTextFocusSuccess: B,
    textColorTextDisabledSuccess: u,
    textColorGhostSuccess: _,
    textColorGhostHoverSuccess: B,
    textColorGhostPressedSuccess: k,
    textColorGhostFocusSuccess: B,
    textColorGhostDisabledSuccess: _,
    borderSuccess: `1px solid ${_}`,
    borderHoverSuccess: `1px solid ${B}`,
    borderPressedSuccess: `1px solid ${k}`,
    borderFocusSuccess: `1px solid ${B}`,
    borderDisabledSuccess: `1px solid ${_}`,
    rippleColorSuccess: _,
    // warning
    colorWarning: v,
    colorHoverWarning: H,
    colorPressedWarning: M,
    colorFocusWarning: H,
    colorDisabledWarning: v,
    textColorWarning: m,
    textColorHoverWarning: m,
    textColorPressedWarning: m,
    textColorFocusWarning: m,
    textColorDisabledWarning: m,
    textColorTextWarning: v,
    textColorTextHoverWarning: H,
    textColorTextPressedWarning: M,
    textColorTextFocusWarning: H,
    textColorTextDisabledWarning: u,
    textColorGhostWarning: v,
    textColorGhostHoverWarning: H,
    textColorGhostPressedWarning: M,
    textColorGhostFocusWarning: H,
    textColorGhostDisabledWarning: v,
    borderWarning: `1px solid ${v}`,
    borderHoverWarning: `1px solid ${H}`,
    borderPressedWarning: `1px solid ${M}`,
    borderFocusWarning: `1px solid ${H}`,
    borderDisabledWarning: `1px solid ${v}`,
    rippleColorWarning: v,
    // error
    colorError: E,
    colorHoverError: W,
    colorPressedError: q,
    colorFocusError: W,
    colorDisabledError: E,
    textColorError: m,
    textColorHoverError: m,
    textColorPressedError: m,
    textColorFocusError: m,
    textColorDisabledError: m,
    textColorTextError: E,
    textColorTextHoverError: W,
    textColorTextPressedError: q,
    textColorTextFocusError: W,
    textColorTextDisabledError: u,
    textColorGhostError: E,
    textColorGhostHoverError: W,
    textColorGhostPressedError: q,
    textColorGhostFocusError: W,
    textColorGhostDisabledError: E,
    borderError: `1px solid ${E}`,
    borderHoverError: `1px solid ${W}`,
    borderPressedError: `1px solid ${q}`,
    borderFocusError: `1px solid ${W}`,
    borderDisabledError: `1px solid ${E}`,
    rippleColorError: E,
    waveOpacity: "0.6",
    fontWeight: Pe,
    fontWeightStrong: Y
  });
}, ks = {
  name: "Button",
  common: C,
  self(e) {
    const o = zs(e);
    return o.waveOpacity = "0.8", o.colorOpacitySecondary = "0.16", o.colorOpacitySecondaryHover = "0.2", o.colorOpacitySecondaryPressed = "0.12", o;
  }
}, U = ks, _s = {
  titleFontSize: "22px"
}, Ms = (e) => {
  const { borderRadius: o, fontSize: r, lineHeight: t, textColor2: n, textColor1: i, textColorDisabled: a, dividerColor: l, fontWeightStrong: s, primaryColor: c, baseColor: d, hoverColor: u, cardColor: g, modalColor: f, popoverColor: p } = e;
  return Object.assign(Object.assign({}, _s), {
    borderRadius: o,
    borderColor: P(g, l),
    borderColorModal: P(f, l),
    borderColorPopover: P(p, l),
    textColor: n,
    titleFontWeight: s,
    titleTextColor: i,
    dayTextColor: a,
    fontSize: r,
    lineHeight: t,
    dateColorCurrent: c,
    dateTextColorCurrent: d,
    cellColorHover: P(g, u),
    cellColorHoverModal: P(f, u),
    cellColorHoverPopover: P(p, u),
    cellColor: g,
    cellColorModal: f,
    cellColorPopover: p,
    barColor: c
  });
}, Os = {
  name: "Calendar",
  common: C,
  peers: {
    Button: U
  },
  self: Ms
}, Bs = Os, As = (e) => {
  const { fontSize: o, boxShadow2: r, popoverColor: t, textColor2: n, borderRadius: i, borderColor: a, heightSmall: l, heightMedium: s, heightLarge: c, fontSizeSmall: d, fontSizeMedium: u, fontSizeLarge: g, dividerColor: f } = e;
  return {
    panelFontSize: o,
    boxShadow: r,
    color: t,
    textColor: n,
    borderRadius: i,
    border: `1px solid ${a}`,
    heightSmall: l,
    heightMedium: s,
    heightLarge: c,
    fontSizeSmall: d,
    fontSizeMedium: u,
    fontSizeLarge: g,
    dividerColor: f
  };
}, Fs = {
  name: "ColorPicker",
  common: C,
  peers: {
    Input: Q,
    Button: U
  },
  self: As
}, Es = Fs, Ws = {
  paddingSmall: "12px 16px 12px",
  paddingMedium: "19px 24px 20px",
  paddingLarge: "23px 32px 24px",
  paddingHuge: "27px 40px 28px",
  titleFontSizeSmall: "16px",
  titleFontSizeMedium: "18px",
  titleFontSizeLarge: "18px",
  titleFontSizeHuge: "18px",
  closeIconSize: "18px",
  closeSize: "22px"
}, Ls = (e) => {
  const { primaryColor: o, borderRadius: r, lineHeight: t, fontSize: n, cardColor: i, textColor2: a, textColor1: l, dividerColor: s, fontWeightStrong: c, closeIconColor: d, closeIconColorHover: u, closeIconColorPressed: g, closeColorHover: f, closeColorPressed: p, modalColor: h, boxShadow1: x, popoverColor: m, actionColor: y } = e;
  return Object.assign(Object.assign({}, Ws), {
    lineHeight: t,
    color: i,
    colorModal: h,
    colorPopover: m,
    colorTarget: o,
    colorEmbedded: y,
    colorEmbeddedModal: y,
    colorEmbeddedPopover: y,
    textColor: a,
    titleTextColor: l,
    borderColor: s,
    actionColor: y,
    titleFontWeight: c,
    closeColorHover: f,
    closeColorPressed: p,
    closeBorderRadius: r,
    closeIconColor: d,
    closeIconColorHover: u,
    closeIconColorPressed: g,
    fontSizeSmall: n,
    fontSizeMedium: n,
    fontSizeLarge: n,
    fontSizeHuge: n,
    boxShadow: x,
    borderRadius: r
  });
}, Rs = {
  name: "Card",
  common: C,
  self(e) {
    const o = Ls(e), { cardColor: r, modalColor: t, popoverColor: n } = e;
    return o.colorEmbedded = r, o.colorEmbeddedModal = t, o.colorEmbeddedPopover = n, o;
  }
}, lt = Rs, js = (e) => ({
  dotSize: "8px",
  dotColor: "rgba(255, 255, 255, .3)",
  dotColorActive: "rgba(255, 255, 255, 1)",
  dotColorFocus: "rgba(255, 255, 255, .5)",
  dotLineWidth: "16px",
  dotLineWidthActive: "24px",
  arrowColor: "#eee"
}), Vs = {
  name: "Carousel",
  common: C,
  self: js
}, Ns = Vs, Gs = {
  sizeSmall: "14px",
  sizeMedium: "16px",
  sizeLarge: "18px",
  labelPadding: "0 8px",
  labelFontWeight: "400"
}, Us = (e) => {
  const { baseColor: o, inputColorDisabled: r, cardColor: t, modalColor: n, popoverColor: i, textColorDisabled: a, borderColor: l, primaryColor: s, textColor2: c, fontSizeSmall: d, fontSizeMedium: u, fontSizeLarge: g, borderRadiusSmall: f, lineHeight: p } = e;
  return Object.assign(Object.assign({}, Gs), {
    labelLineHeight: p,
    fontSizeSmall: d,
    fontSizeMedium: u,
    fontSizeLarge: g,
    borderRadius: f,
    color: o,
    colorChecked: s,
    colorDisabled: r,
    colorDisabledChecked: r,
    colorTableHeader: t,
    colorTableHeaderModal: n,
    colorTableHeaderPopover: i,
    checkMarkColor: o,
    checkMarkColorDisabled: a,
    checkMarkColorDisabledChecked: a,
    border: `1px solid ${l}`,
    borderDisabled: `1px solid ${l}`,
    borderDisabledChecked: `1px solid ${l}`,
    borderChecked: `1px solid ${s}`,
    borderFocus: `1px solid ${s}`,
    boxShadowFocus: `0 0 0 2px ${S(s, { alpha: 0.3 })}`,
    textColor: c,
    textColorDisabled: a
  });
}, qs = {
  name: "Checkbox",
  common: C,
  self(e) {
    const { cardColor: o } = e, r = Us(e);
    return r.color = "#0000", r.checkMarkColor = o, r;
  }
}, $e = qs, Ks = (e) => {
  const { borderRadius: o, boxShadow2: r, popoverColor: t, textColor2: n, textColor3: i, primaryColor: a, textColorDisabled: l, dividerColor: s, hoverColor: c, fontSizeMedium: d, heightMedium: u } = e;
  return {
    menuBorderRadius: o,
    menuColor: t,
    menuBoxShadow: r,
    menuDividerColor: s,
    menuHeight: "calc(var(--n-option-height) * 6.6)",
    optionArrowColor: i,
    optionHeight: u,
    optionFontSize: d,
    optionColorHover: c,
    optionTextColor: n,
    optionTextColorActive: a,
    optionTextColorDisabled: l,
    optionCheckMarkColor: a,
    loadingColor: a,
    columnWidth: "180px"
  };
}, Qs = {
  name: "Cascader",
  common: C,
  peers: {
    InternalSelectMenu: Fe,
    InternalSelection: _o,
    Scrollbar: G,
    Checkbox: $e,
    Empty: Vl
  },
  self: Ks
}, Js = Qs, Ys = {
  name: "Code",
  common: C,
  self(e) {
    const { textColor2: o, fontSize: r, fontWeightStrong: t, textColor3: n } = e;
    return {
      textColor: o,
      fontSize: r,
      fontWeightStrong: t,
      // extracted from hljs atom-one-dark.scss
      "mono-3": "#5c6370",
      "hue-1": "#56b6c2",
      "hue-2": "#61aeee",
      "hue-3": "#c678dd",
      "hue-4": "#98c379",
      "hue-5": "#e06c75",
      "hue-5-2": "#be5046",
      "hue-6": "#d19a66",
      "hue-6-2": "#e6c07b",
      // line-number styles
      lineNumberTextColor: n
    };
  }
}, st = Ys, Xs = (e) => {
  const { fontWeight: o, textColor1: r, textColor2: t, textColorDisabled: n, dividerColor: i, fontSize: a } = e;
  return {
    titleFontSize: a,
    titleFontWeight: o,
    dividerColor: i,
    titleTextColor: r,
    titleTextColorDisabled: n,
    fontSize: a,
    textColor: t,
    arrowColor: t,
    arrowColorDisabled: n,
    itemMargin: "16px 0 0 0",
    titlePadding: "16px 0 0 0"
  };
}, Zs = {
  name: "Collapse",
  common: C,
  self: Xs
}, ec = Zs, oc = (e) => {
  const { cubicBezierEaseInOut: o } = e;
  return {
    bezier: o
  };
}, rc = {
  name: "CollapseTransition",
  common: C,
  self: oc
}, tc = rc, nc = {
  name: "Popselect",
  common: C,
  peers: {
    Popover: fe,
    InternalSelectMenu: Fe
  }
}, ct = nc;
function ic(e) {
  const { boxShadow2: o } = e;
  return {
    menuBoxShadow: o
  };
}
const ac = {
  name: "Select",
  common: C,
  peers: {
    InternalSelection: _o,
    InternalSelectMenu: Fe
  },
  self: ic
}, dt = ac, lc = {
  itemPaddingSmall: "0 4px",
  itemMarginSmall: "0 0 0 8px",
  itemMarginSmallRtl: "0 8px 0 0",
  itemPaddingMedium: "0 4px",
  itemMarginMedium: "0 0 0 8px",
  itemMarginMediumRtl: "0 8px 0 0",
  itemPaddingLarge: "0 4px",
  itemMarginLarge: "0 0 0 8px",
  itemMarginLargeRtl: "0 8px 0 0",
  buttonIconSizeSmall: "14px",
  buttonIconSizeMedium: "16px",
  buttonIconSizeLarge: "18px",
  inputWidthSmall: "60px",
  selectWidthSmall: "unset",
  inputMarginSmall: "0 0 0 8px",
  inputMarginSmallRtl: "0 8px 0 0",
  selectMarginSmall: "0 0 0 8px",
  prefixMarginSmall: "0 8px 0 0",
  suffixMarginSmall: "0 0 0 8px",
  inputWidthMedium: "60px",
  selectWidthMedium: "unset",
  inputMarginMedium: "0 0 0 8px",
  inputMarginMediumRtl: "0 8px 0 0",
  selectMarginMedium: "0 0 0 8px",
  prefixMarginMedium: "0 8px 0 0",
  suffixMarginMedium: "0 0 0 8px",
  inputWidthLarge: "60px",
  selectWidthLarge: "unset",
  inputMarginLarge: "0 0 0 8px",
  inputMarginLargeRtl: "0 8px 0 0",
  selectMarginLarge: "0 0 0 8px",
  prefixMarginLarge: "0 8px 0 0",
  suffixMarginLarge: "0 0 0 8px"
}, sc = (e) => {
  const {
    textColor2: o,
    primaryColor: r,
    primaryColorHover: t,
    primaryColorPressed: n,
    inputColorDisabled: i,
    textColorDisabled: a,
    borderColor: l,
    borderRadius: s,
    // item font size
    fontSizeTiny: c,
    fontSizeSmall: d,
    fontSizeMedium: u,
    // item size
    heightTiny: g,
    heightSmall: f,
    heightMedium: p
  } = e;
  return Object.assign(Object.assign({}, lc), { buttonColor: "#0000", buttonColorHover: "#0000", buttonColorPressed: "#0000", buttonBorder: `1px solid ${l}`, buttonBorderHover: `1px solid ${l}`, buttonBorderPressed: `1px solid ${l}`, buttonIconColor: o, buttonIconColorHover: o, buttonIconColorPressed: o, itemTextColor: o, itemTextColorHover: t, itemTextColorPressed: n, itemTextColorActive: r, itemTextColorDisabled: a, itemColor: "#0000", itemColorHover: "#0000", itemColorPressed: "#0000", itemColorActive: "#0000", itemColorActiveHover: "#0000", itemColorDisabled: i, itemBorder: "1px solid #0000", itemBorderHover: "1px solid #0000", itemBorderPressed: "1px solid #0000", itemBorderActive: `1px solid ${r}`, itemBorderDisabled: `1px solid ${l}`, itemBorderRadius: s, itemSizeSmall: g, itemSizeMedium: f, itemSizeLarge: p, itemFontSizeSmall: c, itemFontSizeMedium: d, itemFontSizeLarge: u, jumperFontSizeSmall: c, jumperFontSizeMedium: d, jumperFontSizeLarge: u, jumperTextColor: o, jumperTextColorDisabled: a });
}, cc = {
  name: "Pagination",
  common: C,
  peers: {
    Select: dt,
    Input: Q,
    Popselect: ct
  },
  self(e) {
    const { primaryColor: o, opacity3: r } = e, t = S(o, {
      alpha: Number(r)
    }), n = sc(e);
    return n.itemBorderActive = `1px solid ${t}`, n.itemBorderDisabled = "1px solid #0000", n;
  }
}, pt = cc, dc = {
  padding: "8px 14px"
}, pc = {
  name: "Tooltip",
  common: C,
  peers: {
    Popover: fe
  },
  self(e) {
    const { borderRadius: o, boxShadow2: r, popoverColor: t, textColor2: n } = e;
    return Object.assign(Object.assign({}, dc), { borderRadius: o, boxShadow: r, color: t, textColor: n });
  }
}, no = pc, uc = {
  name: "Ellipsis",
  common: C,
  peers: {
    Tooltip: no
  }
}, ut = uc, fc = {
  radioSizeSmall: "14px",
  radioSizeMedium: "16px",
  radioSizeLarge: "18px",
  labelPadding: "0 8px",
  labelFontWeight: "400"
}, gc = {
  name: "Radio",
  common: C,
  self(e) {
    const { borderColor: o, primaryColor: r, baseColor: t, textColorDisabled: n, inputColorDisabled: i, textColor2: a, opacityDisabled: l, borderRadius: s, fontSizeSmall: c, fontSizeMedium: d, fontSizeLarge: u, heightSmall: g, heightMedium: f, heightLarge: p, lineHeight: h } = e;
    return Object.assign(Object.assign({}, fc), {
      labelLineHeight: h,
      buttonHeightSmall: g,
      buttonHeightMedium: f,
      buttonHeightLarge: p,
      fontSizeSmall: c,
      fontSizeMedium: d,
      fontSizeLarge: u,
      boxShadow: `inset 0 0 0 1px ${o}`,
      boxShadowActive: `inset 0 0 0 1px ${r}`,
      boxShadowFocus: `inset 0 0 0 1px ${r}, 0 0 0 2px ${S(r, { alpha: 0.3 })}`,
      boxShadowHover: `inset 0 0 0 1px ${r}`,
      boxShadowDisabled: `inset 0 0 0 1px ${o}`,
      color: "#0000",
      colorDisabled: i,
      colorActive: "#0000",
      textColor: a,
      textColorDisabled: n,
      dotColorActive: r,
      dotColorDisabled: o,
      buttonBorderColor: o,
      buttonBorderColorActive: r,
      buttonBorderColorHover: r,
      buttonColor: "#0000",
      buttonColorActive: r,
      buttonTextColor: a,
      buttonTextColorActive: t,
      buttonTextColorHover: r,
      opacityDisabled: l,
      buttonBoxShadowFocus: `inset 0 0 0 1px ${r}, 0 0 0 2px ${S(r, { alpha: 0.3 })}`,
      buttonBoxShadowHover: `inset 0 0 0 1px ${r}`,
      buttonBoxShadow: "inset 0 0 0 1px #0000",
      buttonBorderRadius: s
    });
  }
}, ft = gc, mc = {
  padding: "4px 0",
  optionIconSizeSmall: "14px",
  optionIconSizeMedium: "16px",
  optionIconSizeLarge: "16px",
  optionIconSizeHuge: "18px",
  optionSuffixWidthSmall: "14px",
  optionSuffixWidthMedium: "14px",
  optionSuffixWidthLarge: "16px",
  optionSuffixWidthHuge: "16px",
  optionIconSuffixWidthSmall: "32px",
  optionIconSuffixWidthMedium: "32px",
  optionIconSuffixWidthLarge: "36px",
  optionIconSuffixWidthHuge: "36px",
  optionPrefixWidthSmall: "14px",
  optionPrefixWidthMedium: "14px",
  optionPrefixWidthLarge: "16px",
  optionPrefixWidthHuge: "16px",
  optionIconPrefixWidthSmall: "36px",
  optionIconPrefixWidthMedium: "36px",
  optionIconPrefixWidthLarge: "40px",
  optionIconPrefixWidthHuge: "40px"
}, Cc = (e) => {
  const { primaryColor: o, textColor2: r, dividerColor: t, hoverColor: n, popoverColor: i, invertedColor: a, borderRadius: l, fontSizeSmall: s, fontSizeMedium: c, fontSizeLarge: d, fontSizeHuge: u, heightSmall: g, heightMedium: f, heightLarge: p, heightHuge: h, textColor3: x, opacityDisabled: m } = e;
  return Object.assign(Object.assign({}, mc), {
    optionHeightSmall: g,
    optionHeightMedium: f,
    optionHeightLarge: p,
    optionHeightHuge: h,
    borderRadius: l,
    fontSizeSmall: s,
    fontSizeMedium: c,
    fontSizeLarge: d,
    fontSizeHuge: u,
    // non-inverted
    optionTextColor: r,
    optionTextColorHover: r,
    optionTextColorActive: o,
    optionTextColorChildActive: o,
    color: i,
    dividerColor: t,
    suffixColor: r,
    prefixColor: r,
    optionColorHover: n,
    optionColorActive: S(o, { alpha: 0.1 }),
    groupHeaderTextColor: x,
    // inverted
    optionTextColorInverted: "#BBB",
    optionTextColorHoverInverted: "#FFF",
    optionTextColorActiveInverted: "#FFF",
    optionTextColorChildActiveInverted: "#FFF",
    colorInverted: a,
    dividerColorInverted: "#BBB",
    suffixColorInverted: "#BBB",
    prefixColorInverted: "#BBB",
    optionColorHoverInverted: o,
    optionColorActiveInverted: o,
    groupHeaderTextColorInverted: "#AAA",
    optionOpacityDisabled: m
  });
}, hc = {
  name: "Dropdown",
  common: C,
  peers: {
    Popover: fe
  },
  self(e) {
    const { primaryColorSuppl: o, primaryColor: r, popoverColor: t } = e, n = Cc(e);
    return n.colorInverted = t, n.optionColorActive = S(r, { alpha: 0.15 }), n.optionColorActiveInverted = o, n.optionColorHoverInverted = o, n;
  }
}, Mo = hc, xc = {
  thPaddingSmall: "8px",
  thPaddingMedium: "12px",
  thPaddingLarge: "12px",
  tdPaddingSmall: "8px",
  tdPaddingMedium: "12px",
  tdPaddingLarge: "12px",
  sorterSize: "15px",
  resizableContainerSize: "8px",
  resizableSize: "2px",
  filterSize: "15px",
  paginationMargin: "12px 0 0 0",
  emptyPadding: "48px 0",
  actionPadding: "8px 12px",
  actionButtonMargin: "0 8px 0 0"
}, bc = (e) => {
  const { cardColor: o, modalColor: r, popoverColor: t, textColor2: n, textColor1: i, tableHeaderColor: a, tableColorHover: l, iconColor: s, primaryColor: c, fontWeightStrong: d, borderRadius: u, lineHeight: g, fontSizeSmall: f, fontSizeMedium: p, fontSizeLarge: h, dividerColor: x, heightSmall: m, opacityDisabled: y, tableColorStriped: w } = e;
  return Object.assign(Object.assign({}, xc), {
    actionDividerColor: x,
    lineHeight: g,
    borderRadius: u,
    fontSizeSmall: f,
    fontSizeMedium: p,
    fontSizeLarge: h,
    borderColor: P(o, x),
    tdColorHover: P(o, l),
    tdColorStriped: P(o, w),
    thColor: P(o, a),
    thColorHover: P(P(o, a), l),
    tdColor: o,
    tdTextColor: n,
    thTextColor: i,
    thFontWeight: d,
    thButtonColorHover: l,
    thIconColor: s,
    thIconColorActive: c,
    // modal
    borderColorModal: P(r, x),
    tdColorHoverModal: P(r, l),
    tdColorStripedModal: P(r, w),
    thColorModal: P(r, a),
    thColorHoverModal: P(P(r, a), l),
    tdColorModal: r,
    // popover
    borderColorPopover: P(t, x),
    tdColorHoverPopover: P(t, l),
    tdColorStripedPopover: P(t, w),
    thColorPopover: P(t, a),
    thColorHoverPopover: P(P(t, a), l),
    tdColorPopover: t,
    boxShadowBefore: "inset -12px 0 8px -12px rgba(0, 0, 0, .18)",
    boxShadowAfter: "inset 12px 0 8px -12px rgba(0, 0, 0, .18)",
    // loading
    loadingColor: c,
    loadingSize: m,
    opacityLoading: y
  });
}, vc = {
  name: "DataTable",
  common: C,
  peers: {
    Button: U,
    Checkbox: $e,
    Radio: ft,
    Pagination: pt,
    Scrollbar: G,
    Empty: ue,
    Popover: fe,
    Ellipsis: ut,
    Dropdown: Mo
  },
  self(e) {
    const o = bc(e);
    return o.boxShadowAfter = "inset 12px 0 8px -12px rgba(0, 0, 0, .36)", o.boxShadowBefore = "inset -12px 0 8px -12px rgba(0, 0, 0, .36)", o;
  }
}, Sc = vc, gt = (e) => {
  const { textColorBase: o, opacity1: r, opacity2: t, opacity3: n, opacity4: i, opacity5: a } = e;
  return {
    color: o,
    opacity1Depth: r,
    opacity2Depth: t,
    opacity3Depth: n,
    opacity4Depth: i,
    opacity5Depth: a
  };
}, yc = {
  name: "Icon",
  common: ko,
  self: gt
}, $c = yc, Pc = {
  name: "Icon",
  common: C,
  self: gt
}, wc = Pc, Hc = Ge("icon", `
 height: 1em;
 width: 1em;
 line-height: 1em;
 text-align: center;
 display: inline-block;
 position: relative;
 fill: currentColor;
 transform: translateZ(0);
`, [J("color-transition", {
  transition: "color .3s var(--n-bezier)"
}), J("depth", {
  color: "var(--n-color)"
}, [j("svg", {
  opacity: "var(--n-opacity)",
  transition: "opacity .3s var(--n-bezier)"
})]), j("svg", {
  height: "1em",
  width: "1em"
})]), Dc = Object.assign(Object.assign({}, Ae.props), { depth: [String, Number], size: [Number, String], color: String, component: Object }), Je = te({
  _n_icon__: !0,
  name: "Icon",
  inheritAttrs: !1,
  props: Dc,
  setup(e) {
    const { mergedClsPrefixRef: o, inlineThemeDisabled: r } = Zr(e), t = Ae("Icon", "-icon", Hc, $c, e, o), n = F(() => {
      const { depth: a } = e, { common: { cubicBezierEaseInOut: l }, self: s } = t.value;
      if (a !== void 0) {
        const { color: c, [`opacity${a}Depth`]: d } = s;
        return {
          "--n-bezier": l,
          "--n-color": c,
          "--n-opacity": d
        };
      }
      return {
        "--n-bezier": l,
        "--n-color": "",
        "--n-opacity": ""
      };
    }), i = r ? et("icon", F(() => `${e.depth || "d"}`), n, e) : void 0;
    return {
      mergedClsPrefix: o,
      mergedStyle: F(() => {
        const { size: a, color: l } = e;
        return {
          fontSize: tn(a),
          color: l
        };
      }),
      cssVars: r ? void 0 : n,
      themeClass: i == null ? void 0 : i.themeClass,
      onRender: i == null ? void 0 : i.onRender
    };
  },
  render() {
    var e;
    const { $parent: o, depth: r, mergedClsPrefix: t, component: n, onRender: i, themeClass: a } = this;
    return !((e = o == null ? void 0 : o.$options) === null || e === void 0) && e._n_icon__ && on("icon", "don't wrap `n-icon` inside `n-icon`"), i == null || i(), I("i", Rt(this.$attrs, {
      role: "img",
      class: [
        `${t}-icon`,
        a,
        {
          [`${t}-icon--depth`]: r,
          [`${t}-icon--color-transition`]: r !== void 0
        }
      ],
      style: [this.cssVars, this.mergedStyle]
    }), n ? I(n) : this.$slots);
  }
}), Tc = {
  itemFontSize: "12px",
  itemHeight: "36px",
  itemWidth: "52px",
  panelActionPadding: "8px 0"
}, Ic = (e) => {
  const { popoverColor: o, textColor2: r, primaryColor: t, hoverColor: n, dividerColor: i, opacityDisabled: a, boxShadow2: l, borderRadius: s, iconColor: c, iconColorDisabled: d } = e;
  return Object.assign(Object.assign({}, Tc), {
    panelColor: o,
    panelBoxShadow: l,
    panelDividerColor: i,
    itemTextColor: r,
    itemTextColorActive: t,
    itemColorHover: n,
    itemOpacityDisabled: a,
    itemBorderRadius: s,
    borderRadius: s,
    iconColor: c,
    iconColorDisabled: d
  });
}, zc = {
  name: "TimePicker",
  common: C,
  peers: {
    Scrollbar: G,
    Button: U,
    Input: Q
  },
  self: Ic
}, mt = zc, kc = {
  itemSize: "24px",
  itemCellWidth: "38px",
  itemCellHeight: "32px",
  scrollItemWidth: "80px",
  scrollItemHeight: "40px",
  panelExtraFooterPadding: "8px 12px",
  panelActionPadding: "8px 12px",
  calendarTitlePadding: "0",
  calendarTitleHeight: "28px",
  arrowSize: "14px",
  panelHeaderPadding: "8px 12px",
  calendarDaysHeight: "32px",
  calendarTitleGridTempateColumns: "28px 28px 1fr 28px 28px",
  // type
  calendarLeftPaddingDate: "6px 12px 4px 12px",
  calendarLeftPaddingDatetime: "4px 12px",
  calendarLeftPaddingDaterange: "6px 12px 4px 12px",
  calendarLeftPaddingDatetimerange: "4px 12px",
  calendarLeftPaddingMonth: "0",
  calendarLeftPaddingYear: "0",
  calendarLeftPaddingQuarter: "0",
  calendarLeftPaddingMonthrange: "0",
  calendarLeftPaddingQuarterrange: "0",
  calendarLeftPaddingYearrange: "0",
  calendarRightPaddingDate: "6px 12px 4px 12px",
  calendarRightPaddingDatetime: "4px 12px",
  calendarRightPaddingDaterange: "6px 12px 4px 12px",
  calendarRightPaddingDatetimerange: "4px 12px",
  calendarRightPaddingMonth: "0",
  calendarRightPaddingYear: "0",
  calendarRightPaddingQuarter: "0",
  calendarRightPaddingMonthrange: "0",
  calendarRightPaddingQuarterrange: "0",
  calendarRightPaddingYearrange: "0"
}, _c = (e) => {
  const { hoverColor: o, fontSize: r, textColor2: t, textColorDisabled: n, popoverColor: i, primaryColor: a, borderRadiusSmall: l, iconColor: s, iconColorDisabled: c, textColor1: d, dividerColor: u, boxShadow2: g, borderRadius: f, fontWeightStrong: p } = e;
  return Object.assign(Object.assign({}, kc), {
    itemFontSize: r,
    calendarDaysFontSize: r,
    calendarTitleFontSize: r,
    itemTextColor: t,
    itemTextColorDisabled: n,
    itemTextColorActive: i,
    itemTextColorCurrent: a,
    itemColorIncluded: S(a, { alpha: 0.1 }),
    itemColorHover: o,
    itemColorDisabled: o,
    itemColorActive: a,
    itemBorderRadius: l,
    panelColor: i,
    panelTextColor: t,
    arrowColor: s,
    calendarTitleTextColor: d,
    calendarTitleColorHover: o,
    calendarDaysTextColor: t,
    panelHeaderDividerColor: u,
    calendarDaysDividerColor: u,
    calendarDividerColor: u,
    panelActionDividerColor: u,
    panelBoxShadow: g,
    panelBorderRadius: f,
    calendarTitleFontWeight: p,
    scrollItemBorderRadius: f,
    iconColor: s,
    iconColorDisabled: c
  });
}, Mc = {
  name: "DatePicker",
  common: C,
  peers: {
    Input: Q,
    Button: U,
    TimePicker: mt,
    Scrollbar: G
  },
  self(e) {
    const { popoverColor: o, hoverColor: r, primaryColor: t } = e, n = _c(e);
    return n.itemColorDisabled = P(o, r), n.itemColorIncluded = S(t, { alpha: 0.15 }), n.itemColorHover = P(o, r), n;
  }
}, Oc = Mc, Bc = {
  thPaddingBorderedSmall: "8px 12px",
  thPaddingBorderedMedium: "12px 16px",
  thPaddingBorderedLarge: "16px 24px",
  thPaddingSmall: "0",
  thPaddingMedium: "0",
  thPaddingLarge: "0",
  tdPaddingBorderedSmall: "8px 12px",
  tdPaddingBorderedMedium: "12px 16px",
  tdPaddingBorderedLarge: "16px 24px",
  tdPaddingSmall: "0 0 8px 0",
  tdPaddingMedium: "0 0 12px 0",
  tdPaddingLarge: "0 0 16px 0"
}, Ac = (e) => {
  const { tableHeaderColor: o, textColor2: r, textColor1: t, cardColor: n, modalColor: i, popoverColor: a, dividerColor: l, borderRadius: s, fontWeightStrong: c, lineHeight: d, fontSizeSmall: u, fontSizeMedium: g, fontSizeLarge: f } = e;
  return Object.assign(Object.assign({}, Bc), {
    lineHeight: d,
    fontSizeSmall: u,
    fontSizeMedium: g,
    fontSizeLarge: f,
    titleTextColor: t,
    thColor: P(n, o),
    thColorModal: P(i, o),
    thColorPopover: P(a, o),
    thTextColor: t,
    thFontWeight: c,
    tdTextColor: r,
    tdColor: n,
    tdColorModal: i,
    tdColorPopover: a,
    borderColor: P(n, l),
    borderColorModal: P(i, l),
    borderColorPopover: P(a, l),
    borderRadius: s
  });
}, Fc = {
  name: "Descriptions",
  common: C,
  self: Ac
}, Ec = Fc, Wc = {
  titleFontSize: "18px",
  padding: "16px 28px 20px 28px",
  iconSize: "28px",
  actionSpace: "12px",
  contentMargin: "8px 0 16px 0",
  iconMargin: "0 4px 0 0",
  iconMarginIconTop: "4px 0 8px 0",
  closeSize: "22px",
  closeIconSize: "18px",
  closeMargin: "20px 26px 0 0",
  closeMarginIconTop: "10px 16px 0 0"
}, Lc = (e) => {
  const { textColor1: o, textColor2: r, modalColor: t, closeIconColor: n, closeIconColorHover: i, closeIconColorPressed: a, closeColorHover: l, closeColorPressed: s, infoColor: c, successColor: d, warningColor: u, errorColor: g, primaryColor: f, dividerColor: p, borderRadius: h, fontWeightStrong: x, lineHeight: m, fontSize: y } = e;
  return Object.assign(Object.assign({}, Wc), {
    fontSize: y,
    lineHeight: m,
    border: `1px solid ${p}`,
    titleTextColor: o,
    textColor: r,
    color: t,
    closeColorHover: l,
    closeColorPressed: s,
    closeIconColor: n,
    closeIconColorHover: i,
    closeIconColorPressed: a,
    closeBorderRadius: h,
    iconColor: f,
    iconColorInfo: c,
    iconColorSuccess: d,
    iconColorWarning: u,
    iconColorError: g,
    borderRadius: h,
    titleFontWeight: x
  });
}, Rc = {
  name: "Dialog",
  common: C,
  peers: {
    Button: U
  },
  self: Lc
}, Ct = Rc, jc = (e) => {
  const { modalColor: o, textColor2: r, boxShadow3: t } = e;
  return {
    color: o,
    textColor: r,
    boxShadow: t
  };
}, Vc = {
  name: "Modal",
  common: C,
  peers: {
    Scrollbar: G,
    Dialog: Ct,
    Card: lt
  },
  self: jc
}, Nc = Vc, Gc = (e) => {
  const { textColor1: o, dividerColor: r, fontWeightStrong: t } = e;
  return {
    textColor: o,
    color: r,
    fontWeight: t
  };
}, Uc = {
  name: "Divider",
  common: C,
  self: Gc
}, qc = Uc, Kc = (e) => {
  const { modalColor: o, textColor1: r, textColor2: t, boxShadow3: n, lineHeight: i, fontWeightStrong: a, dividerColor: l, closeColorHover: s, closeColorPressed: c, closeIconColor: d, closeIconColorHover: u, closeIconColorPressed: g, borderRadius: f, primaryColorHover: p } = e;
  return {
    bodyPadding: "16px 24px",
    headerPadding: "16px 24px",
    footerPadding: "16px 24px",
    color: o,
    textColor: t,
    titleTextColor: r,
    titleFontSize: "18px",
    titleFontWeight: a,
    boxShadow: n,
    lineHeight: i,
    headerBorderBottom: `1px solid ${l}`,
    footerBorderTop: `1px solid ${l}`,
    closeIconColor: d,
    closeIconColorHover: u,
    closeIconColorPressed: g,
    closeSize: "22px",
    closeIconSize: "18px",
    closeColorHover: s,
    closeColorPressed: c,
    closeBorderRadius: f,
    resizableTriggerColorHover: p
  };
}, Qc = {
  name: "Drawer",
  common: C,
  peers: {
    Scrollbar: G
  },
  self: Kc
}, Jc = Qc, Yc = {
  actionMargin: "0 0 0 20px",
  actionMarginRtl: "0 20px 0 0"
}, Xc = {
  name: "DynamicInput",
  common: C,
  peers: {
    Input: Q,
    Button: U
  },
  self() {
    return Yc;
  }
}, Zc = Xc, ed = {
  gapSmall: "4px 8px",
  gapMedium: "8px 12px",
  gapLarge: "12px 16px"
}, od = {
  name: "Space",
  self() {
    return ed;
  }
}, ht = od, rd = {
  name: "DynamicTags",
  common: C,
  peers: {
    Input: Q,
    Button: U,
    Tag: it,
    Space: ht
  },
  self() {
    return {
      inputWidth: "64px"
    };
  }
}, td = rd, nd = {
  name: "Element",
  common: C
}, id = nd, ad = {
  feedbackPadding: "4px 0 0 2px",
  feedbackHeightSmall: "24px",
  feedbackHeightMedium: "24px",
  feedbackHeightLarge: "26px",
  feedbackFontSizeSmall: "13px",
  feedbackFontSizeMedium: "14px",
  feedbackFontSizeLarge: "14px",
  labelFontSizeLeftSmall: "14px",
  labelFontSizeLeftMedium: "14px",
  labelFontSizeLeftLarge: "15px",
  labelFontSizeTopSmall: "13px",
  labelFontSizeTopMedium: "14px",
  labelFontSizeTopLarge: "14px",
  labelHeightSmall: "24px",
  labelHeightMedium: "26px",
  labelHeightLarge: "28px",
  labelPaddingVertical: "0 0 6px 2px",
  labelPaddingHorizontal: "0 12px 0 0",
  labelTextAlignVertical: "left",
  labelTextAlignHorizontal: "right",
  labelFontWeight: "400"
}, ld = (e) => {
  const { heightSmall: o, heightMedium: r, heightLarge: t, textColor1: n, errorColor: i, warningColor: a, lineHeight: l, textColor3: s } = e;
  return Object.assign(Object.assign({}, ad), { blankHeightSmall: o, blankHeightMedium: r, blankHeightLarge: t, lineHeight: l, labelTextColor: n, asteriskColor: i, feedbackTextColorError: i, feedbackTextColorWarning: a, feedbackTextColor: s });
}, sd = {
  name: "Form",
  common: C,
  self: ld
}, cd = sd, dd = {
  name: "GradientText",
  common: C,
  self(e) {
    const { primaryColor: o, successColor: r, warningColor: t, errorColor: n, infoColor: i, primaryColorSuppl: a, successColorSuppl: l, warningColorSuppl: s, errorColorSuppl: c, infoColorSuppl: d, fontWeightStrong: u } = e;
    return {
      fontWeight: u,
      rotate: "252deg",
      colorStartPrimary: o,
      colorEndPrimary: a,
      colorStartInfo: i,
      colorEndInfo: d,
      colorStartWarning: t,
      colorEndWarning: s,
      colorStartError: n,
      colorEndError: c,
      colorStartSuccess: r,
      colorEndSuccess: l
    };
  }
}, pd = dd, ud = (e) => {
  const { primaryColor: o, baseColor: r } = e;
  return {
    color: o,
    iconColor: r
  };
}, fd = {
  name: "IconWrapper",
  common: C,
  self: ud
}, gd = fd, md = {
  closeMargin: "16px 12px",
  closeSize: "20px",
  closeIconSize: "16px",
  width: "365px",
  padding: "16px",
  titleFontSize: "16px",
  metaFontSize: "12px",
  descriptionFontSize: "12px"
}, Cd = (e) => {
  const { textColor2: o, successColor: r, infoColor: t, warningColor: n, errorColor: i, popoverColor: a, closeIconColor: l, closeIconColorHover: s, closeIconColorPressed: c, closeColorHover: d, closeColorPressed: u, textColor1: g, textColor3: f, borderRadius: p, fontWeightStrong: h, boxShadow2: x, lineHeight: m, fontSize: y } = e;
  return Object.assign(Object.assign({}, md), {
    borderRadius: p,
    lineHeight: m,
    fontSize: y,
    headerFontWeight: h,
    iconColor: o,
    iconColorSuccess: r,
    iconColorInfo: t,
    iconColorWarning: n,
    iconColorError: i,
    color: a,
    textColor: o,
    closeIconColor: l,
    closeIconColorHover: s,
    closeIconColorPressed: c,
    closeBorderRadius: p,
    closeColorHover: d,
    closeColorPressed: u,
    headerTextColor: g,
    descriptionTextColor: f,
    actionTextColor: o,
    boxShadow: x
  });
}, hd = {
  name: "Notification",
  common: C,
  peers: {
    Scrollbar: G
  },
  self: Cd
}, xd = hd, bd = {
  margin: "0 0 8px 0",
  padding: "10px 20px",
  maxWidth: "720px",
  minWidth: "420px",
  iconMargin: "0 10px 0 0",
  closeMargin: "0 0 0 10px",
  closeSize: "20px",
  closeIconSize: "16px",
  iconSize: "20px",
  fontSize: "14px"
}, vd = (e) => {
  const { textColor2: o, closeIconColor: r, closeIconColorHover: t, closeIconColorPressed: n, infoColor: i, successColor: a, errorColor: l, warningColor: s, popoverColor: c, boxShadow2: d, primaryColor: u, lineHeight: g, borderRadius: f, closeColorHover: p, closeColorPressed: h } = e;
  return Object.assign(Object.assign({}, bd), {
    closeBorderRadius: f,
    textColor: o,
    textColorInfo: o,
    textColorSuccess: o,
    textColorError: o,
    textColorWarning: o,
    textColorLoading: o,
    color: c,
    colorInfo: c,
    colorSuccess: c,
    colorError: c,
    colorWarning: c,
    colorLoading: c,
    boxShadow: d,
    boxShadowInfo: d,
    boxShadowSuccess: d,
    boxShadowError: d,
    boxShadowWarning: d,
    boxShadowLoading: d,
    iconColor: o,
    iconColorInfo: i,
    iconColorSuccess: a,
    iconColorWarning: s,
    iconColorError: l,
    iconColorLoading: u,
    closeColorHover: p,
    closeColorPressed: h,
    closeIconColor: r,
    closeIconColorHover: t,
    closeIconColorPressed: n,
    closeColorHoverInfo: p,
    closeColorPressedInfo: h,
    closeIconColorInfo: r,
    closeIconColorHoverInfo: t,
    closeIconColorPressedInfo: n,
    closeColorHoverSuccess: p,
    closeColorPressedSuccess: h,
    closeIconColorSuccess: r,
    closeIconColorHoverSuccess: t,
    closeIconColorPressedSuccess: n,
    closeColorHoverError: p,
    closeColorPressedError: h,
    closeIconColorError: r,
    closeIconColorHoverError: t,
    closeIconColorPressedError: n,
    closeColorHoverWarning: p,
    closeColorPressedWarning: h,
    closeIconColorWarning: r,
    closeIconColorHoverWarning: t,
    closeIconColorPressedWarning: n,
    closeColorHoverLoading: p,
    closeColorPressedLoading: h,
    closeIconColorLoading: r,
    closeIconColorHoverLoading: t,
    closeIconColorPressedLoading: n,
    loadingColor: u,
    lineHeight: g,
    borderRadius: f
  });
}, Sd = {
  name: "Message",
  common: C,
  self: vd
}, yd = Sd, $d = {
  name: "ButtonGroup",
  common: C
}, Pd = $d, wd = {
  name: "InputNumber",
  common: C,
  peers: {
    Button: U,
    Input: Q
  },
  self(e) {
    const { textColorDisabled: o } = e;
    return {
      iconColorDisabled: o
    };
  }
}, Hd = wd, Dd = {
  name: "Layout",
  common: C,
  peers: {
    Scrollbar: G
  },
  self(e) {
    const { textColor2: o, bodyColor: r, popoverColor: t, cardColor: n, dividerColor: i, scrollbarColor: a, scrollbarColorHover: l } = e;
    return {
      textColor: o,
      textColorInverted: o,
      color: r,
      colorEmbedded: r,
      headerColor: n,
      headerColorInverted: n,
      footerColor: n,
      footerColorInverted: n,
      headerBorderColor: i,
      headerBorderColorInverted: i,
      footerBorderColor: i,
      footerBorderColorInverted: i,
      siderBorderColor: i,
      siderBorderColorInverted: i,
      siderColor: n,
      siderColorInverted: n,
      siderToggleButtonBorder: "1px solid transparent",
      siderToggleButtonColor: t,
      siderToggleButtonIconColor: o,
      siderToggleButtonIconColorInverted: o,
      siderToggleBarColor: P(r, a),
      siderToggleBarColorHover: P(r, l),
      __invertScrollbar: "false"
    };
  }
}, Td = Dd, Id = (e) => {
  const { textColor2: o, cardColor: r, modalColor: t, popoverColor: n, dividerColor: i, borderRadius: a, fontSize: l, hoverColor: s } = e;
  return {
    textColor: o,
    color: r,
    colorHover: s,
    colorModal: t,
    colorHoverModal: P(t, s),
    colorPopover: n,
    colorHoverPopover: P(n, s),
    borderColor: i,
    borderColorModal: P(t, i),
    borderColorPopover: P(n, i),
    borderRadius: a,
    fontSize: l
  };
}, zd = {
  name: "List",
  common: C,
  self: Id
}, kd = zd, _d = {
  name: "LoadingBar",
  common: C,
  self(e) {
    const { primaryColor: o } = e;
    return {
      colorError: "red",
      colorLoading: o,
      height: "2px"
    };
  }
}, Md = _d, Od = {
  name: "Log",
  common: C,
  peers: {
    Scrollbar: G,
    Code: st
  },
  self(e) {
    const { textColor2: o, inputColor: r, fontSize: t, primaryColor: n } = e;
    return {
      loaderFontSize: t,
      loaderTextColor: o,
      loaderColor: r,
      loaderBorder: "1px solid #0000",
      loadingColor: n
    };
  }
}, Bd = Od, Ad = {
  name: "Mention",
  common: C,
  peers: {
    InternalSelectMenu: Fe,
    Input: Q
  },
  self(e) {
    const { boxShadow2: o } = e;
    return {
      menuBoxShadow: o
    };
  }
}, Fd = Ad;
function Ed(e, o, r, t) {
  return {
    itemColorHoverInverted: "#0000",
    itemColorActiveInverted: o,
    itemColorActiveHoverInverted: o,
    itemColorActiveCollapsedInverted: o,
    itemTextColorInverted: e,
    itemTextColorHoverInverted: r,
    itemTextColorChildActiveInverted: r,
    itemTextColorChildActiveHoverInverted: r,
    itemTextColorActiveInverted: r,
    itemTextColorActiveHoverInverted: r,
    itemTextColorHorizontalInverted: e,
    itemTextColorHoverHorizontalInverted: r,
    itemTextColorChildActiveHorizontalInverted: r,
    itemTextColorChildActiveHoverHorizontalInverted: r,
    itemTextColorActiveHorizontalInverted: r,
    itemTextColorActiveHoverHorizontalInverted: r,
    itemIconColorInverted: e,
    itemIconColorHoverInverted: r,
    itemIconColorActiveInverted: r,
    itemIconColorActiveHoverInverted: r,
    itemIconColorChildActiveInverted: r,
    itemIconColorChildActiveHoverInverted: r,
    itemIconColorCollapsedInverted: e,
    itemIconColorHorizontalInverted: e,
    itemIconColorHoverHorizontalInverted: r,
    itemIconColorActiveHorizontalInverted: r,
    itemIconColorActiveHoverHorizontalInverted: r,
    itemIconColorChildActiveHorizontalInverted: r,
    itemIconColorChildActiveHoverHorizontalInverted: r,
    arrowColorInverted: e,
    arrowColorHoverInverted: r,
    arrowColorActiveInverted: r,
    arrowColorActiveHoverInverted: r,
    arrowColorChildActiveInverted: r,
    arrowColorChildActiveHoverInverted: r,
    groupTextColorInverted: t
  };
}
const Wd = (e) => {
  const { borderRadius: o, textColor3: r, primaryColor: t, textColor2: n, textColor1: i, fontSize: a, dividerColor: l, hoverColor: s, primaryColorHover: c } = e;
  return Object.assign({
    borderRadius: o,
    color: "#0000",
    groupTextColor: r,
    itemColorHover: s,
    itemColorActive: S(t, { alpha: 0.1 }),
    itemColorActiveHover: S(t, { alpha: 0.1 }),
    itemColorActiveCollapsed: S(t, { alpha: 0.1 }),
    itemTextColor: n,
    itemTextColorHover: n,
    itemTextColorActive: t,
    itemTextColorActiveHover: t,
    itemTextColorChildActive: t,
    itemTextColorChildActiveHover: t,
    itemTextColorHorizontal: n,
    itemTextColorHoverHorizontal: c,
    itemTextColorActiveHorizontal: t,
    itemTextColorActiveHoverHorizontal: t,
    itemTextColorChildActiveHorizontal: t,
    itemTextColorChildActiveHoverHorizontal: t,
    itemIconColor: i,
    itemIconColorHover: i,
    itemIconColorActive: t,
    itemIconColorActiveHover: t,
    itemIconColorChildActive: t,
    itemIconColorChildActiveHover: t,
    itemIconColorCollapsed: i,
    itemIconColorHorizontal: i,
    itemIconColorHoverHorizontal: c,
    itemIconColorActiveHorizontal: t,
    itemIconColorActiveHoverHorizontal: t,
    itemIconColorChildActiveHorizontal: t,
    itemIconColorChildActiveHoverHorizontal: t,
    itemHeight: "42px",
    arrowColor: n,
    arrowColorHover: n,
    arrowColorActive: t,
    arrowColorActiveHover: t,
    arrowColorChildActive: t,
    arrowColorChildActiveHover: t,
    colorInverted: "#0000",
    borderColorHorizontal: "#0000",
    fontSize: a,
    dividerColor: l
  }, Ed("#BBB", t, "#FFF", "#AAA"));
}, Ld = {
  name: "Menu",
  common: C,
  peers: {
    Tooltip: no,
    Dropdown: Mo
  },
  self(e) {
    const { primaryColor: o, primaryColorSuppl: r } = e, t = Wd(e);
    return t.itemColorActive = S(o, { alpha: 0.15 }), t.itemColorActiveHover = S(o, { alpha: 0.15 }), t.itemColorActiveCollapsed = S(o, {
      alpha: 0.15
    }), t.itemColorActiveInverted = r, t.itemColorActiveHoverInverted = r, t.itemColorActiveCollapsedInverted = r, t;
  }
}, Rd = Ld, jd = {
  titleFontSize: "18px",
  backSize: "22px"
};
function Vd(e) {
  const { textColor1: o, textColor2: r, textColor3: t, fontSize: n, fontWeightStrong: i, primaryColorHover: a, primaryColorPressed: l } = e;
  return Object.assign(Object.assign({}, jd), { titleFontWeight: i, fontSize: n, titleTextColor: o, backColor: r, backColorHover: a, backColorPressed: l, subtitleTextColor: t });
}
const Nd = {
  name: "PageHeader",
  common: C,
  self: Vd
}, Gd = {
  iconSize: "22px"
}, Ud = (e) => {
  const { fontSize: o, warningColor: r } = e;
  return Object.assign(Object.assign({}, Gd), { fontSize: o, iconColor: r });
}, qd = {
  name: "Popconfirm",
  common: C,
  peers: {
    Button: U,
    Popover: fe
  },
  self: Ud
}, Kd = qd, Qd = (e) => {
  const { infoColor: o, successColor: r, warningColor: t, errorColor: n, textColor2: i, progressRailColor: a, fontSize: l, fontWeight: s } = e;
  return {
    fontSize: l,
    fontSizeCircle: "28px",
    fontWeightCircle: s,
    railColor: a,
    railHeight: "8px",
    iconSizeCircle: "36px",
    iconSizeLine: "18px",
    iconColor: o,
    iconColorInfo: o,
    iconColorSuccess: r,
    iconColorWarning: t,
    iconColorError: n,
    textColorCircle: i,
    textColorLineInner: "rgb(255, 255, 255)",
    textColorLineOuter: i,
    fillColor: o,
    fillColorInfo: o,
    fillColorSuccess: r,
    fillColorWarning: t,
    fillColorError: n,
    lineBgProcessing: "linear-gradient(90deg, rgba(255, 255, 255, .3) 0%, rgba(255, 255, 255, .5) 100%)"
  };
}, Jd = {
  name: "Progress",
  common: C,
  self(e) {
    const o = Qd(e);
    return o.textColorLineInner = "rgb(0, 0, 0)", o.lineBgProcessing = "linear-gradient(90deg, rgba(255, 255, 255, .3) 0%, rgba(255, 255, 255, .5) 100%)", o;
  }
}, xt = Jd, Yd = {
  name: "Rate",
  common: C,
  self(e) {
    const { railColor: o } = e;
    return {
      itemColor: o,
      itemColorActive: "#CCAA33",
      itemSize: "20px",
      sizeSmall: "16px",
      sizeMedium: "20px",
      sizeLarge: "24px"
    };
  }
}, Xd = Yd, Zd = {
  titleFontSizeSmall: "26px",
  titleFontSizeMedium: "32px",
  titleFontSizeLarge: "40px",
  titleFontSizeHuge: "48px",
  fontSizeSmall: "14px",
  fontSizeMedium: "14px",
  fontSizeLarge: "15px",
  fontSizeHuge: "16px",
  iconSizeSmall: "64px",
  iconSizeMedium: "80px",
  iconSizeLarge: "100px",
  iconSizeHuge: "125px",
  iconColor418: void 0,
  iconColor404: void 0,
  iconColor403: void 0,
  iconColor500: void 0
}, ep = (e) => {
  const { textColor2: o, textColor1: r, errorColor: t, successColor: n, infoColor: i, warningColor: a, lineHeight: l, fontWeightStrong: s } = e;
  return Object.assign(Object.assign({}, Zd), { lineHeight: l, titleFontWeight: s, titleTextColor: r, textColor: o, iconColorError: t, iconColorSuccess: n, iconColorInfo: i, iconColorWarning: a });
}, op = {
  name: "Result",
  common: C,
  self: ep
}, rp = op, tp = {
  railHeight: "4px",
  railWidthVertical: "4px",
  handleSize: "18px",
  dotHeight: "8px",
  dotWidth: "8px",
  dotBorderRadius: "4px"
}, np = {
  name: "Slider",
  common: C,
  self(e) {
    const o = "0 2px 8px 0 rgba(0, 0, 0, 0.12)", { railColor: r, modalColor: t, primaryColorSuppl: n, popoverColor: i, textColor2: a, cardColor: l, borderRadius: s, fontSize: c, opacityDisabled: d } = e;
    return Object.assign(Object.assign({}, tp), { fontSize: c, markFontSize: c, railColor: r, railColorHover: r, fillColor: n, fillColorHover: n, opacityDisabled: d, handleColor: "#FFF", dotColor: l, dotColorModal: t, dotColorPopover: i, handleBoxShadow: "0px 2px 4px 0 rgba(0, 0, 0, 0.4)", handleBoxShadowHover: "0px 2px 4px 0 rgba(0, 0, 0, 0.4)", handleBoxShadowActive: "0px 2px 4px 0 rgba(0, 0, 0, 0.4)", handleBoxShadowFocus: "0px 2px 4px 0 rgba(0, 0, 0, 0.4)", indicatorColor: i, indicatorBoxShadow: o, indicatorTextColor: a, indicatorBorderRadius: s, dotBorder: `2px solid ${r}`, dotBorderActive: `2px solid ${n}`, dotBoxShadow: "" });
  }
}, ip = np, ap = (e) => {
  const { opacityDisabled: o, heightTiny: r, heightSmall: t, heightMedium: n, heightLarge: i, heightHuge: a, primaryColor: l, fontSize: s } = e;
  return {
    fontSize: s,
    textColor: l,
    sizeTiny: r,
    sizeSmall: t,
    sizeMedium: n,
    sizeLarge: i,
    sizeHuge: a,
    color: l,
    opacitySpinning: o
  };
}, lp = {
  name: "Spin",
  common: C,
  self: ap
}, sp = lp, cp = (e) => {
  const { textColor2: o, textColor3: r, fontSize: t, fontWeight: n } = e;
  return {
    labelFontSize: t,
    labelFontWeight: n,
    valueFontWeight: n,
    valueFontSize: "24px",
    labelTextColor: r,
    valuePrefixTextColor: o,
    valueSuffixTextColor: o,
    valueTextColor: o
  };
}, dp = {
  name: "Statistic",
  common: C,
  self: cp
}, pp = dp, up = {
  stepHeaderFontSizeSmall: "14px",
  stepHeaderFontSizeMedium: "16px",
  indicatorIndexFontSizeSmall: "14px",
  indicatorIndexFontSizeMedium: "16px",
  indicatorSizeSmall: "22px",
  indicatorSizeMedium: "28px",
  indicatorIconSizeSmall: "14px",
  indicatorIconSizeMedium: "18px"
}, fp = (e) => {
  const { fontWeightStrong: o, baseColor: r, textColorDisabled: t, primaryColor: n, errorColor: i, textColor1: a, textColor2: l } = e;
  return Object.assign(Object.assign({}, up), { stepHeaderFontWeight: o, indicatorTextColorProcess: r, indicatorTextColorWait: t, indicatorTextColorFinish: n, indicatorTextColorError: i, indicatorBorderColorProcess: n, indicatorBorderColorWait: t, indicatorBorderColorFinish: n, indicatorBorderColorError: i, indicatorColorProcess: n, indicatorColorWait: "#0000", indicatorColorFinish: "#0000", indicatorColorError: "#0000", splitorColorProcess: t, splitorColorWait: t, splitorColorFinish: n, splitorColorError: t, headerTextColorProcess: a, headerTextColorWait: t, headerTextColorFinish: t, headerTextColorError: i, descriptionTextColorProcess: l, descriptionTextColorWait: t, descriptionTextColorFinish: t, descriptionTextColorError: i });
}, gp = {
  name: "Steps",
  common: C,
  self: fp
}, mp = gp, bt = {
  buttonHeightSmall: "14px",
  buttonHeightMedium: "18px",
  buttonHeightLarge: "22px",
  buttonWidthSmall: "14px",
  buttonWidthMedium: "18px",
  buttonWidthLarge: "22px",
  buttonWidthPressedSmall: "20px",
  buttonWidthPressedMedium: "24px",
  buttonWidthPressedLarge: "28px",
  railHeightSmall: "18px",
  railHeightMedium: "22px",
  railHeightLarge: "26px",
  railWidthSmall: "32px",
  railWidthMedium: "40px",
  railWidthLarge: "48px"
}, Cp = {
  name: "Switch",
  common: C,
  self(e) {
    const { primaryColorSuppl: o, opacityDisabled: r, borderRadius: t, primaryColor: n, textColor2: i, baseColor: a } = e, l = "rgba(255, 255, 255, .20)";
    return Object.assign(Object.assign({}, bt), { iconColor: a, textColor: i, loadingColor: o, opacityDisabled: r, railColor: l, railColorActive: o, buttonBoxShadow: "0px 2px 4px 0 rgba(0, 0, 0, 0.4)", buttonColor: "#FFF", railBorderRadiusSmall: t, railBorderRadiusMedium: t, railBorderRadiusLarge: t, buttonBorderRadiusSmall: t, buttonBorderRadiusMedium: t, buttonBorderRadiusLarge: t, boxShadowFocus: `0 0 8px 0 ${S(n, { alpha: 0.3 })}` });
  }
}, hp = Cp, xp = (e) => {
  const { primaryColor: o, opacityDisabled: r, borderRadius: t, textColor3: n } = e, i = "rgba(0, 0, 0, .14)";
  return Object.assign(Object.assign({}, bt), { iconColor: n, textColor: "white", loadingColor: o, opacityDisabled: r, railColor: i, railColorActive: o, buttonBoxShadow: "0 1px 4px 0 rgba(0, 0, 0, 0.3), inset 0 0 1px 0 rgba(0, 0, 0, 0.05)", buttonColor: "#FFF", railBorderRadiusSmall: t, railBorderRadiusMedium: t, railBorderRadiusLarge: t, buttonBorderRadiusSmall: t, buttonBorderRadiusMedium: t, buttonBorderRadiusLarge: t, boxShadowFocus: `0 0 0 2px ${S(o, { alpha: 0.2 })}` });
}, bp = {
  name: "Switch",
  common: ko,
  self: xp
}, vp = bp, Sp = {
  thPaddingSmall: "6px",
  thPaddingMedium: "12px",
  thPaddingLarge: "12px",
  tdPaddingSmall: "6px",
  tdPaddingMedium: "12px",
  tdPaddingLarge: "12px"
}, yp = (e) => {
  const { dividerColor: o, cardColor: r, modalColor: t, popoverColor: n, tableHeaderColor: i, tableColorStriped: a, textColor1: l, textColor2: s, borderRadius: c, fontWeightStrong: d, lineHeight: u, fontSizeSmall: g, fontSizeMedium: f, fontSizeLarge: p } = e;
  return Object.assign(Object.assign({}, Sp), {
    fontSizeSmall: g,
    fontSizeMedium: f,
    fontSizeLarge: p,
    lineHeight: u,
    borderRadius: c,
    borderColor: P(r, o),
    borderColorModal: P(t, o),
    borderColorPopover: P(n, o),
    tdColor: r,
    tdColorModal: t,
    tdColorPopover: n,
    tdColorStriped: P(r, a),
    tdColorStripedModal: P(t, a),
    tdColorStripedPopover: P(n, a),
    thColor: P(r, i),
    thColorModal: P(t, i),
    thColorPopover: P(n, i),
    thTextColor: l,
    tdTextColor: s,
    thFontWeight: d
  });
}, $p = {
  name: "Table",
  common: C,
  self: yp
}, Pp = $p, wp = {
  tabFontSizeSmall: "14px",
  tabFontSizeMedium: "14px",
  tabFontSizeLarge: "16px",
  tabGapSmallLine: "36px",
  tabGapMediumLine: "36px",
  tabGapLargeLine: "36px",
  tabGapSmallLineVertical: "8px",
  tabGapMediumLineVertical: "8px",
  tabGapLargeLineVertical: "8px",
  tabPaddingSmallLine: "6px 0",
  tabPaddingMediumLine: "10px 0",
  tabPaddingLargeLine: "14px 0",
  tabPaddingVerticalSmallLine: "6px 12px",
  tabPaddingVerticalMediumLine: "8px 16px",
  tabPaddingVerticalLargeLine: "10px 20px",
  tabGapSmallBar: "36px",
  tabGapMediumBar: "36px",
  tabGapLargeBar: "36px",
  tabGapSmallBarVertical: "8px",
  tabGapMediumBarVertical: "8px",
  tabGapLargeBarVertical: "8px",
  tabPaddingSmallBar: "4px 0",
  tabPaddingMediumBar: "6px 0",
  tabPaddingLargeBar: "10px 0",
  tabPaddingVerticalSmallBar: "6px 12px",
  tabPaddingVerticalMediumBar: "8px 16px",
  tabPaddingVerticalLargeBar: "10px 20px",
  tabGapSmallCard: "4px",
  tabGapMediumCard: "4px",
  tabGapLargeCard: "4px",
  tabGapSmallCardVertical: "4px",
  tabGapMediumCardVertical: "4px",
  tabGapLargeCardVertical: "4px",
  tabPaddingSmallCard: "8px 16px",
  tabPaddingMediumCard: "10px 20px",
  tabPaddingLargeCard: "12px 24px",
  tabPaddingSmallSegment: "4px 0",
  tabPaddingMediumSegment: "6px 0",
  tabPaddingLargeSegment: "8px 0",
  tabPaddingVerticalLargeSegment: "0 8px",
  tabPaddingVerticalSmallCard: "8px 12px",
  tabPaddingVerticalMediumCard: "10px 16px",
  tabPaddingVerticalLargeCard: "12px 20px",
  tabPaddingVerticalSmallSegment: "0 4px",
  tabPaddingVerticalMediumSegment: "0 6px",
  tabGapSmallSegment: "0",
  tabGapMediumSegment: "0",
  tabGapLargeSegment: "0",
  tabGapSmallSegmentVertical: "0",
  tabGapMediumSegmentVertical: "0",
  tabGapLargeSegmentVertical: "0",
  panePaddingSmall: "8px 0 0 0",
  panePaddingMedium: "12px 0 0 0",
  panePaddingLarge: "16px 0 0 0",
  closeSize: "18px",
  closeIconSize: "14px"
}, Hp = (e) => {
  const { textColor2: o, primaryColor: r, textColorDisabled: t, closeIconColor: n, closeIconColorHover: i, closeIconColorPressed: a, closeColorHover: l, closeColorPressed: s, tabColor: c, baseColor: d, dividerColor: u, fontWeight: g, textColor1: f, borderRadius: p, fontSize: h, fontWeightStrong: x } = e;
  return Object.assign(Object.assign({}, wp), {
    colorSegment: c,
    tabFontSizeCard: h,
    tabTextColorLine: f,
    tabTextColorActiveLine: r,
    tabTextColorHoverLine: r,
    tabTextColorDisabledLine: t,
    tabTextColorSegment: f,
    tabTextColorActiveSegment: o,
    tabTextColorHoverSegment: o,
    tabTextColorDisabledSegment: t,
    tabTextColorBar: f,
    tabTextColorActiveBar: r,
    tabTextColorHoverBar: r,
    tabTextColorDisabledBar: t,
    tabTextColorCard: f,
    tabTextColorHoverCard: f,
    tabTextColorActiveCard: r,
    tabTextColorDisabledCard: t,
    barColor: r,
    closeIconColor: n,
    closeIconColorHover: i,
    closeIconColorPressed: a,
    closeColorHover: l,
    closeColorPressed: s,
    closeBorderRadius: p,
    tabColor: c,
    tabColorSegment: d,
    tabBorderColor: u,
    tabFontWeightActive: g,
    tabFontWeight: g,
    tabBorderRadius: p,
    paneTextColor: o,
    fontWeightStrong: x
  });
}, Dp = {
  name: "Tabs",
  common: C,
  self(e) {
    const o = Hp(e), { inputColor: r } = e;
    return o.colorSegment = r, o.tabColorSegment = r, o;
  }
}, Tp = Dp, Ip = (e) => {
  const { textColor1: o, textColor2: r, fontWeightStrong: t, fontSize: n } = e;
  return {
    fontSize: n,
    titleTextColor: o,
    textColor: r,
    titleFontWeight: t
  };
}, zp = {
  name: "Thing",
  common: C,
  self: Ip
}, kp = zp, _p = {
  titleMarginMedium: "0 0 6px 0",
  titleMarginLarge: "-2px 0 6px 0",
  titleFontSizeMedium: "14px",
  titleFontSizeLarge: "16px",
  iconSizeMedium: "14px",
  iconSizeLarge: "14px"
}, Mp = {
  name: "Timeline",
  common: C,
  self(e) {
    const { textColor3: o, infoColorSuppl: r, errorColorSuppl: t, successColorSuppl: n, warningColorSuppl: i, textColor1: a, textColor2: l, railColor: s, fontWeightStrong: c, fontSize: d } = e;
    return Object.assign(Object.assign({}, _p), { contentFontSize: d, titleFontWeight: c, circleBorder: `2px solid ${o}`, circleBorderInfo: `2px solid ${r}`, circleBorderError: `2px solid ${t}`, circleBorderSuccess: `2px solid ${n}`, circleBorderWarning: `2px solid ${i}`, iconColor: o, iconColorInfo: r, iconColorError: t, iconColorSuccess: n, iconColorWarning: i, titleTextColor: a, contentTextColor: l, metaTextColor: o, lineColor: s });
  }
}, Op = Mp, Bp = {
  extraFontSizeSmall: "12px",
  extraFontSizeMedium: "12px",
  extraFontSizeLarge: "14px",
  titleFontSizeSmall: "14px",
  titleFontSizeMedium: "16px",
  titleFontSizeLarge: "16px",
  closeSize: "20px",
  closeIconSize: "16px",
  headerHeightSmall: "44px",
  headerHeightMedium: "44px",
  headerHeightLarge: "50px"
}, Ap = {
  name: "Transfer",
  common: C,
  peers: {
    Checkbox: $e,
    Scrollbar: G,
    Input: Q,
    Empty: ue,
    Button: U
  },
  self(e) {
    const { fontWeight: o, fontSizeLarge: r, fontSizeMedium: t, fontSizeSmall: n, heightLarge: i, heightMedium: a, borderRadius: l, inputColor: s, tableHeaderColor: c, textColor1: d, textColorDisabled: u, textColor2: g, textColor3: f, hoverColor: p, closeColorHover: h, closeColorPressed: x, closeIconColor: m, closeIconColorHover: y, closeIconColorPressed: w, dividerColor: z } = e;
    return Object.assign(Object.assign({}, Bp), {
      itemHeightSmall: a,
      itemHeightMedium: a,
      itemHeightLarge: i,
      fontSizeSmall: n,
      fontSizeMedium: t,
      fontSizeLarge: r,
      borderRadius: l,
      dividerColor: z,
      borderColor: "#0000",
      listColor: s,
      headerColor: c,
      titleTextColor: d,
      titleTextColorDisabled: u,
      extraTextColor: f,
      extraTextColorDisabled: u,
      itemTextColor: g,
      itemTextColorDisabled: u,
      itemColorPending: p,
      titleFontWeight: o,
      closeColorHover: h,
      closeColorPressed: x,
      closeIconColor: m,
      closeIconColorHover: y,
      closeIconColorPressed: w
    });
  }
}, Fp = Ap, Ep = (e) => {
  const { borderRadiusSmall: o, hoverColor: r, pressedColor: t, primaryColor: n, textColor3: i, textColor2: a, textColorDisabled: l, fontSize: s } = e;
  return {
    fontSize: s,
    nodeBorderRadius: o,
    nodeColorHover: r,
    nodeColorPressed: t,
    nodeColorActive: S(n, { alpha: 0.1 }),
    arrowColor: i,
    nodeTextColor: a,
    nodeTextColorDisabled: l,
    loadingColor: n,
    dropMarkColor: n
  };
}, Wp = {
  name: "Tree",
  common: C,
  peers: {
    Checkbox: $e,
    Scrollbar: G,
    Empty: ue
  },
  self(e) {
    const { primaryColor: o } = e, r = Ep(e);
    return r.nodeColorActive = S(o, { alpha: 0.15 }), r;
  }
}, vt = Wp, Lp = {
  name: "TreeSelect",
  common: C,
  peers: {
    Tree: vt,
    Empty: ue,
    InternalSelection: _o
  }
}, Rp = Lp, jp = {
  headerFontSize1: "30px",
  headerFontSize2: "22px",
  headerFontSize3: "18px",
  headerFontSize4: "16px",
  headerFontSize5: "16px",
  headerFontSize6: "16px",
  headerMargin1: "28px 0 20px 0",
  headerMargin2: "28px 0 20px 0",
  headerMargin3: "28px 0 20px 0",
  headerMargin4: "28px 0 18px 0",
  headerMargin5: "28px 0 18px 0",
  headerMargin6: "28px 0 18px 0",
  headerPrefixWidth1: "16px",
  headerPrefixWidth2: "16px",
  headerPrefixWidth3: "12px",
  headerPrefixWidth4: "12px",
  headerPrefixWidth5: "12px",
  headerPrefixWidth6: "12px",
  headerBarWidth1: "4px",
  headerBarWidth2: "4px",
  headerBarWidth3: "3px",
  headerBarWidth4: "3px",
  headerBarWidth5: "3px",
  headerBarWidth6: "3px",
  pMargin: "16px 0 16px 0",
  liMargin: ".25em 0 0 0",
  olPadding: "0 0 0 2em",
  ulPadding: "0 0 0 2em"
}, Vp = (e) => {
  const { primaryColor: o, textColor2: r, borderColor: t, lineHeight: n, fontSize: i, borderRadiusSmall: a, dividerColor: l, fontWeightStrong: s, textColor1: c, textColor3: d, infoColor: u, warningColor: g, errorColor: f, successColor: p, codeColor: h } = e;
  return Object.assign(Object.assign({}, jp), { aTextColor: o, blockquoteTextColor: r, blockquotePrefixColor: t, blockquoteLineHeight: n, blockquoteFontSize: i, codeBorderRadius: a, liTextColor: r, liLineHeight: n, liFontSize: i, hrColor: l, headerFontWeight: s, headerTextColor: c, pTextColor: r, pTextColor1Depth: c, pTextColor2Depth: r, pTextColor3Depth: d, pLineHeight: n, pFontSize: i, headerBarColor: o, headerBarColorPrimary: o, headerBarColorInfo: u, headerBarColorError: f, headerBarColorWarning: g, headerBarColorSuccess: p, textColor: r, textColor1Depth: c, textColor2Depth: r, textColor3Depth: d, textColorPrimary: o, textColorInfo: u, textColorSuccess: p, textColorWarning: g, textColorError: f, codeTextColor: r, codeColor: h, codeBorder: "1px solid #0000" });
}, Np = {
  name: "Typography",
  common: C,
  self: Vp
}, Gp = Np, Up = (e) => {
  const { iconColor: o, primaryColor: r, errorColor: t, textColor2: n, successColor: i, opacityDisabled: a, actionColor: l, borderColor: s, hoverColor: c, lineHeight: d, borderRadius: u, fontSize: g } = e;
  return {
    fontSize: g,
    lineHeight: d,
    borderRadius: u,
    draggerColor: l,
    draggerBorder: `1px dashed ${s}`,
    draggerBorderHover: `1px dashed ${r}`,
    itemColorHover: c,
    itemColorHoverError: S(t, {
      alpha: 0.06
    }),
    itemTextColor: n,
    itemTextColorError: t,
    itemTextColorSuccess: i,
    itemIconColor: o,
    itemDisabledOpacity: a,
    itemBorderImageCardError: `1px solid ${t}`,
    itemBorderImageCard: `1px solid ${s}`
  };
}, qp = {
  name: "Upload",
  common: C,
  peers: {
    Button: U,
    Progress: xt
  },
  self(e) {
    const { errorColor: o } = e, r = Up(e);
    return r.itemColorHoverError = S(o, {
      alpha: 0.09
    }), r;
  }
}, Kp = qp, Qp = {
  name: "Watermark",
  common: C,
  self(e) {
    const { fontFamily: o } = e;
    return {
      fontFamily: o
    };
  }
}, Jp = Qp, Yp = {
  name: "Row",
  common: C
}, Xp = Yp, Zp = {
  name: "Image",
  common: C,
  peers: {
    Tooltip: no
  },
  self: (e) => {
    const { textColor2: o } = e;
    return {
      toolbarIconColor: o,
      toolbarColor: "rgba(0, 0, 0, .35)",
      toolbarBoxShadow: "none",
      toolbarBorderRadius: "24px"
    };
  }
}, eu = {
  extraFontSize: "12px",
  width: "440px"
}, ou = {
  name: "Transfer",
  common: C,
  peers: {
    Checkbox: $e,
    Scrollbar: G,
    Input: Q,
    Empty: ue,
    Button: U
  },
  self(e) {
    const { iconColorDisabled: o, iconColor: r, fontWeight: t, fontSizeLarge: n, fontSizeMedium: i, fontSizeSmall: a, heightLarge: l, heightMedium: s, heightSmall: c, borderRadius: d, inputColor: u, tableHeaderColor: g, textColor1: f, textColorDisabled: p, textColor2: h, hoverColor: x } = e;
    return Object.assign(Object.assign({}, eu), {
      itemHeightSmall: c,
      itemHeightMedium: s,
      itemHeightLarge: l,
      fontSizeSmall: a,
      fontSizeMedium: i,
      fontSizeLarge: n,
      borderRadius: d,
      borderColor: "#0000",
      listColor: u,
      headerColor: g,
      titleTextColor: f,
      titleTextColorDisabled: p,
      extraTextColor: h,
      filterDividerColor: "#0000",
      itemTextColor: h,
      itemTextColorDisabled: p,
      itemColorPending: x,
      titleFontWeight: t,
      iconColor: r,
      iconColorDisabled: o
    });
  }
}, ru = ou, tu = {
  name: "Skeleton",
  common: C,
  self(e) {
    const { heightSmall: o, heightMedium: r, heightLarge: t, borderRadius: n } = e;
    return {
      color: "rgba(255, 255, 255, 0.12)",
      colorEnd: "rgba(255, 255, 255, 0.18)",
      borderRadius: n,
      heightSmall: o,
      heightMedium: r,
      heightLarge: t
    };
  }
}, nu = Ge("switch", `
 height: var(--n-height);
 min-width: var(--n-width);
 vertical-align: middle;
 user-select: none;
 -webkit-user-select: none;
 display: inline-flex;
 outline: none;
 justify-content: center;
 align-items: center;
`, [D("children-placeholder", `
 height: var(--n-rail-height);
 display: flex;
 flex-direction: column;
 overflow: hidden;
 pointer-events: none;
 visibility: hidden;
 `), D("rail-placeholder", `
 display: flex;
 flex-wrap: none;
 `), D("button-placeholder", `
 width: calc(1.75 * var(--n-rail-height));
 height: var(--n-rail-height);
 `), Ge("base-loading", `
 position: absolute;
 top: 50%;
 left: 50%;
 transform: translateX(-50%) translateY(-50%);
 font-size: calc(var(--n-button-width) - 4px);
 color: var(--n-loading-color);
 transition: color .3s var(--n-bezier);
 `, [Qe({
  left: "50%",
  top: "50%",
  originalTransform: "translateX(-50%) translateY(-50%)"
})]), D("checked, unchecked", `
 transition: color .3s var(--n-bezier);
 color: var(--n-text-color);
 box-sizing: border-box;
 position: absolute;
 white-space: nowrap;
 top: 0;
 bottom: 0;
 display: flex;
 align-items: center;
 line-height: 1;
 `), D("checked", `
 right: 0;
 padding-right: calc(1.25 * var(--n-rail-height) - var(--n-offset));
 `), D("unchecked", `
 left: 0;
 justify-content: flex-end;
 padding-left: calc(1.25 * var(--n-rail-height) - var(--n-offset));
 `), j("&:focus", [D("rail", `
 box-shadow: var(--n-box-shadow-focus);
 `)]), J("round", [D("rail", "border-radius: calc(var(--n-rail-height) / 2);", [D("button", "border-radius: calc(var(--n-button-height) / 2);")])]), Lo("disabled", [Lo("icon", [J("rubber-band", [J("pressed", [D("rail", [D("button", "max-width: var(--n-button-width-pressed);")])]), D("rail", [j("&:active", [D("button", "max-width: var(--n-button-width-pressed);")])]), J("active", [J("pressed", [D("rail", [D("button", "left: calc(100% - var(--n-offset) - var(--n-button-width-pressed));")])]), D("rail", [j("&:active", [D("button", "left: calc(100% - var(--n-offset) - var(--n-button-width-pressed));")])])])])])]), J("active", [D("rail", [D("button", "left: calc(100% - var(--n-button-width) - var(--n-offset))")])]), D("rail", `
 overflow: hidden;
 height: var(--n-rail-height);
 min-width: var(--n-rail-width);
 border-radius: var(--n-rail-border-radius);
 cursor: pointer;
 position: relative;
 transition:
 opacity .3s var(--n-bezier),
 background .3s var(--n-bezier),
 box-shadow .3s var(--n-bezier);
 background-color: var(--n-rail-color);
 `, [D("button-icon", `
 color: var(--n-icon-color);
 transition: color .3s var(--n-bezier);
 font-size: calc(var(--n-button-height) - 4px);
 position: absolute;
 left: 0;
 right: 0;
 top: 0;
 bottom: 0;
 display: flex;
 justify-content: center;
 align-items: center;
 line-height: 1;
 `, [Qe()]), D("button", `
 align-items: center; 
 top: var(--n-offset);
 left: var(--n-offset);
 height: var(--n-button-height);
 width: var(--n-button-width-pressed);
 max-width: var(--n-button-width);
 border-radius: var(--n-button-border-radius);
 background-color: var(--n-button-color);
 box-shadow: var(--n-button-box-shadow);
 box-sizing: border-box;
 cursor: inherit;
 content: "";
 position: absolute;
 transition:
 background-color .3s var(--n-bezier),
 left .3s var(--n-bezier),
 opacity .3s var(--n-bezier),
 max-width .3s var(--n-bezier),
 box-shadow .3s var(--n-bezier);
 `)]), J("active", [D("rail", "background-color: var(--n-rail-color-active);")]), J("loading", [D("rail", `
 cursor: wait;
 `)]), J("disabled", [D("rail", `
 cursor: not-allowed;
 opacity: .5;
 `)])]), iu = Object.assign(Object.assign({}, Ae.props), {
  size: {
    type: String,
    default: "medium"
  },
  value: {
    type: [String, Number, Boolean],
    default: void 0
  },
  loading: Boolean,
  defaultValue: {
    type: [String, Number, Boolean],
    default: !1
  },
  disabled: {
    type: Boolean,
    default: void 0
  },
  round: {
    type: Boolean,
    default: !0
  },
  "onUpdate:value": [Function, Array],
  onUpdateValue: [Function, Array],
  checkedValue: {
    type: [String, Number, Boolean],
    default: !0
  },
  uncheckedValue: {
    type: [String, Number, Boolean],
    default: !1
  },
  railStyle: Function,
  rubberBand: {
    type: Boolean,
    default: !0
  },
  /** @deprecated */
  onChange: [Function, Array]
});
let Ie;
const au = te({
  name: "Switch",
  props: iu,
  setup(e) {
    process.env.NODE_ENV !== "production" && $o(() => {
      e.onChange && en("switch", "`on-change` is deprecated, please use `on-update:value` instead.");
    }), Ie === void 0 && (typeof CSS < "u" ? typeof CSS.supports < "u" ? Ie = CSS.supports("width", "max(1px)") : Ie = !1 : Ie = !0);
    const { mergedClsPrefixRef: o, inlineThemeDisabled: r } = Zr(e), t = Ae("Switch", "-switch", nu, vp, e, o), n = kn(e), { mergedSizeRef: i, mergedDisabledRef: a } = n, l = K(e.defaultValue), s = yr(e, "value"), c = Hn(s, l), d = F(() => c.value === e.checkedValue), u = K(!1), g = K(!1), f = F(() => {
      const { railStyle: v } = e;
      if (v)
        return v({ focused: g.value, checked: d.value });
    });
    function p(v) {
      const { "onUpdate:value": H, onChange: M, onUpdateValue: E } = e, { nTriggerFormInput: W, nTriggerFormChange: q } = n;
      H && Ve(H, v), E && Ve(E, v), M && Ve(M, v), l.value = v, W(), q();
    }
    function h() {
      const { nTriggerFormFocus: v } = n;
      v();
    }
    function x() {
      const { nTriggerFormBlur: v } = n;
      v();
    }
    function m() {
      e.loading || a.value || (c.value !== e.checkedValue ? p(e.checkedValue) : p(e.uncheckedValue));
    }
    function y() {
      g.value = !0, h();
    }
    function w() {
      g.value = !1, x(), u.value = !1;
    }
    function z(v) {
      e.loading || a.value || v.key === " " && (c.value !== e.checkedValue ? p(e.checkedValue) : p(e.uncheckedValue), u.value = !1);
    }
    function _(v) {
      e.loading || a.value || v.key === " " && (v.preventDefault(), u.value = !0);
    }
    const B = F(() => {
      const { value: v } = i, { self: { opacityDisabled: H, railColor: M, railColorActive: E, buttonBoxShadow: W, buttonColor: q, boxShadowFocus: Pe, loadingColor: Ee, textColor: we, iconColor: He, [re("buttonHeight", v)]: Y, [re("buttonWidth", v)]: kt, [re("buttonWidthPressed", v)]: _t, [re("railHeight", v)]: oe, [re("railWidth", v)]: De, [re("railBorderRadius", v)]: Mt, [re("buttonBorderRadius", v)]: Ot }, common: { cubicBezierEaseInOut: Bt } } = t.value;
      let io, ao, lo;
      return Ie ? (io = `calc((${oe} - ${Y}) / 2)`, ao = `max(${oe}, ${Y})`, lo = `max(${De}, calc(${De} + ${Y} - ${oe}))`) : (io = so((X(oe) - X(Y)) / 2), ao = so(Math.max(X(oe), X(Y))), lo = X(oe) > X(Y) ? De : so(X(De) + X(Y) - X(oe))), {
        "--n-bezier": Bt,
        "--n-button-border-radius": Ot,
        "--n-button-box-shadow": W,
        "--n-button-color": q,
        "--n-button-width": kt,
        "--n-button-width-pressed": _t,
        "--n-button-height": Y,
        "--n-height": ao,
        "--n-offset": io,
        "--n-opacity-disabled": H,
        "--n-rail-border-radius": Mt,
        "--n-rail-color": M,
        "--n-rail-color-active": E,
        "--n-rail-height": oe,
        "--n-rail-width": De,
        "--n-width": lo,
        "--n-box-shadow-focus": Pe,
        "--n-loading-color": Ee,
        "--n-text-color": we,
        "--n-icon-color": He
      };
    }), k = r ? et("switch", F(() => i.value[0]), B, e) : void 0;
    return {
      handleClick: m,
      handleBlur: w,
      handleFocus: y,
      handleKeyup: z,
      handleKeydown: _,
      mergedRailStyle: f,
      pressed: u,
      mergedClsPrefix: o,
      mergedValue: c,
      checked: d,
      mergedDisabled: a,
      cssVars: r ? void 0 : B,
      themeClass: k == null ? void 0 : k.themeClass,
      onRender: k == null ? void 0 : k.onRender
    };
  },
  render() {
    const { mergedClsPrefix: e, mergedDisabled: o, checked: r, mergedRailStyle: t, onRender: n, $slots: i } = this;
    n == null || n();
    const { checked: a, unchecked: l, icon: s, "checked-icon": c, "unchecked-icon": d } = i, u = !(po(s) && po(c) && po(d));
    return I(
      "div",
      { role: "switch", "aria-checked": r, class: [
        `${e}-switch`,
        this.themeClass,
        u && `${e}-switch--icon`,
        r && `${e}-switch--active`,
        o && `${e}-switch--disabled`,
        this.round && `${e}-switch--round`,
        this.loading && `${e}-switch--loading`,
        this.pressed && `${e}-switch--pressed`,
        this.rubberBand && `${e}-switch--rubber-band`
      ], tabindex: this.mergedDisabled ? void 0 : 0, style: this.cssVars, onClick: this.handleClick, onFocus: this.handleFocus, onBlur: this.handleBlur, onKeyup: this.handleKeyup, onKeydown: this.handleKeydown },
      I(
        "div",
        { class: `${e}-switch__rail`, "aria-hidden": "true", style: t },
        ne(a, (g) => ne(l, (f) => g || f ? I(
          "div",
          { "aria-hidden": !0, class: `${e}-switch__children-placeholder` },
          I(
            "div",
            { class: `${e}-switch__rail-placeholder` },
            I("div", { class: `${e}-switch__button-placeholder` }),
            g
          ),
          I(
            "div",
            { class: `${e}-switch__rail-placeholder` },
            I("div", { class: `${e}-switch__button-placeholder` }),
            f
          )
        ) : null)),
        I(
          "div",
          { class: `${e}-switch__button` },
          ne(s, (g) => ne(c, (f) => ne(d, (p) => I(ot, null, {
            default: () => this.loading ? I(Ml, { key: "loading", clsPrefix: e, strokeWidth: 20 }) : this.checked && (f || g) ? I("div", { class: `${e}-switch__button-icon`, key: f ? "checked-icon" : "icon" }, f || g) : !this.checked && (p || g) ? I("div", { class: `${e}-switch__button-icon`, key: p ? "unchecked-icon" : "icon" }, p || g) : null
          })))),
          ne(a, (g) => g && I("div", { key: "checked", class: `${e}-switch__checked` }, g)),
          ne(l, (g) => g && I("div", { key: "unchecked", class: `${e}-switch__unchecked` }, g))
        )
      )
    );
  }
}), lu = () => ({}), su = {
  name: "Equation",
  common: C,
  self: lu
}, cu = su, du = {
  name: "dark",
  common: C,
  Alert: is,
  Anchor: cs,
  AutoComplete: gs,
  Avatar: at,
  AvatarGroup: bs,
  BackTop: ys,
  Badge: Ps,
  Breadcrumb: Ts,
  Button: U,
  ButtonGroup: Pd,
  Calendar: Bs,
  Card: lt,
  Carousel: Ns,
  Cascader: Js,
  Checkbox: $e,
  Code: st,
  Collapse: ec,
  CollapseTransition: tc,
  ColorPicker: Es,
  DataTable: Sc,
  DatePicker: Oc,
  Descriptions: Ec,
  Dialog: Ct,
  Divider: qc,
  Drawer: Jc,
  Dropdown: Mo,
  DynamicInput: Zc,
  DynamicTags: td,
  Element: id,
  Empty: ue,
  Ellipsis: ut,
  Equation: cu,
  Form: cd,
  GradientText: pd,
  Icon: wc,
  IconWrapper: gd,
  Image: Zp,
  Input: Q,
  InputNumber: Hd,
  LegacyTransfer: ru,
  Layout: Td,
  List: kd,
  LoadingBar: Md,
  Log: Bd,
  Menu: Rd,
  Mention: Fd,
  Message: yd,
  Modal: Nc,
  Notification: xd,
  PageHeader: Nd,
  Pagination: pt,
  Popconfirm: Kd,
  Popover: fe,
  Popselect: ct,
  Progress: xt,
  Radio: ft,
  Rate: Xd,
  Result: rp,
  Row: Xp,
  Scrollbar: G,
  Select: dt,
  Skeleton: tu,
  Slider: ip,
  Space: ht,
  Spin: sp,
  Statistic: pp,
  Steps: mp,
  Switch: hp,
  Table: Pp,
  Tabs: Tp,
  Tag: it,
  Thing: kp,
  TimePicker: mt,
  Timeline: Op,
  Tooltip: no,
  Transfer: Fp,
  Tree: vt,
  TreeSelect: Rp,
  Typography: Gp,
  Upload: Kp,
  Watermark: Jp
};
var rr;
const St = typeof window < "u", pu = (e) => typeof e == "function", uu = (e) => typeof e == "string", fu = () => {
};
St && ((rr = window == null ? void 0 : window.navigator) != null && rr.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function Oe(e) {
  return typeof e == "function" ? e() : N(e);
}
function gu(e, o) {
  function r(...t) {
    return new Promise((n, i) => {
      Promise.resolve(e(() => o.apply(this, t), { fn: o, thisArg: this, args: t })).then(n).catch(i);
    });
  }
  return r;
}
const yt = (e) => e();
function mu(e = yt) {
  const o = K(!0);
  function r() {
    o.value = !1;
  }
  function t() {
    o.value = !0;
  }
  const n = (...i) => {
    o.value && e(...i);
  };
  return { isActive: br(o), pause: r, resume: t, eventFilter: n };
}
function Cu(e) {
  return e;
}
function $t(e) {
  return Vt() ? (Nt(e), !0) : !1;
}
function hu(e) {
  return typeof e == "function" ? F(e) : K(e);
}
function Pt(e, o = !0) {
  jt() ? xr(e) : o ? e() : Pr(e);
}
function xu(e = !1, o = {}) {
  const {
    truthyValue: r = !0,
    falsyValue: t = !1
  } = o, n = $r(e), i = K(e);
  function a(l) {
    if (arguments.length)
      return i.value = l, i.value;
    {
      const s = Oe(r);
      return i.value = i.value === s ? Oe(t) : s, i.value;
    }
  }
  return n ? a : [i, a];
}
var tr = Object.getOwnPropertySymbols, bu = Object.prototype.hasOwnProperty, vu = Object.prototype.propertyIsEnumerable, Su = (e, o) => {
  var r = {};
  for (var t in e)
    bu.call(e, t) && o.indexOf(t) < 0 && (r[t] = e[t]);
  if (e != null && tr)
    for (var t of tr(e))
      o.indexOf(t) < 0 && vu.call(e, t) && (r[t] = e[t]);
  return r;
};
function yu(e, o, r = {}) {
  const t = r, {
    eventFilter: n = yt
  } = t, i = Su(t, [
    "eventFilter"
  ]);
  return _e(e, gu(n, o), i);
}
var $u = Object.defineProperty, Pu = Object.defineProperties, wu = Object.getOwnPropertyDescriptors, Ye = Object.getOwnPropertySymbols, wt = Object.prototype.hasOwnProperty, Ht = Object.prototype.propertyIsEnumerable, nr = (e, o, r) => o in e ? $u(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, Hu = (e, o) => {
  for (var r in o || (o = {}))
    wt.call(o, r) && nr(e, r, o[r]);
  if (Ye)
    for (var r of Ye(o))
      Ht.call(o, r) && nr(e, r, o[r]);
  return e;
}, Du = (e, o) => Pu(e, wu(o)), Tu = (e, o) => {
  var r = {};
  for (var t in e)
    wt.call(e, t) && o.indexOf(t) < 0 && (r[t] = e[t]);
  if (e != null && Ye)
    for (var t of Ye(e))
      o.indexOf(t) < 0 && Ht.call(e, t) && (r[t] = e[t]);
  return r;
};
function Iu(e, o, r = {}) {
  const t = r, {
    eventFilter: n
  } = t, i = Tu(t, [
    "eventFilter"
  ]), { eventFilter: a, pause: l, resume: s, isActive: c } = mu(n);
  return { stop: yu(e, o, Du(Hu({}, i), {
    eventFilter: a
  })), pause: l, resume: s, isActive: c };
}
function zu(e) {
  var o;
  const r = Oe(e);
  return (o = r == null ? void 0 : r.$el) != null ? o : r;
}
const he = St ? window : void 0;
function ir(...e) {
  let o, r, t, n;
  if (uu(e[0]) || Array.isArray(e[0]) ? ([r, t, n] = e, o = he) : [o, r, t, n] = e, !o)
    return fu;
  Array.isArray(r) || (r = [r]), Array.isArray(t) || (t = [t]);
  const i = [], a = () => {
    i.forEach((d) => d()), i.length = 0;
  }, l = (d, u, g, f) => (d.addEventListener(u, g, f), () => d.removeEventListener(u, g, f)), s = _e(() => [zu(o), Oe(n)], ([d, u]) => {
    a(), d && i.push(...r.flatMap((g) => t.map((f) => l(d, g, f, u))));
  }, { immediate: !0, flush: "post" }), c = () => {
    s(), a();
  };
  return $t(c), c;
}
function ku(e, o = !1) {
  const r = K(), t = () => r.value = !!e();
  return t(), Pt(t, o), r;
}
function _u(e, o = {}) {
  const { window: r = he } = o, t = ku(() => r && "matchMedia" in r && typeof r.matchMedia == "function");
  let n;
  const i = K(!1), a = () => {
    n && ("removeEventListener" in n ? n.removeEventListener("change", l) : n.removeListener(l));
  }, l = () => {
    t.value && (a(), n = r.matchMedia(hu(e).value), i.value = n.matches, "addEventListener" in n ? n.addEventListener("change", l) : n.addListener(l));
  };
  return $o(l), $t(() => a()), i;
}
const So = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, yo = "__vueuse_ssr_handlers__";
So[yo] = So[yo] || {};
const Mu = So[yo];
function Dt(e, o) {
  return Mu[e] || o;
}
function Ou(e) {
  return e == null ? "any" : e instanceof Set ? "set" : e instanceof Map ? "map" : e instanceof Date ? "date" : typeof e == "boolean" ? "boolean" : typeof e == "string" ? "string" : typeof e == "object" ? "object" : Number.isNaN(e) ? "any" : "number";
}
var Bu = Object.defineProperty, ar = Object.getOwnPropertySymbols, Au = Object.prototype.hasOwnProperty, Fu = Object.prototype.propertyIsEnumerable, lr = (e, o, r) => o in e ? Bu(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, sr = (e, o) => {
  for (var r in o || (o = {}))
    Au.call(o, r) && lr(e, r, o[r]);
  if (ar)
    for (var r of ar(o))
      Fu.call(o, r) && lr(e, r, o[r]);
  return e;
};
const Eu = {
  boolean: {
    read: (e) => e === "true",
    write: (e) => String(e)
  },
  object: {
    read: (e) => JSON.parse(e),
    write: (e) => JSON.stringify(e)
  },
  number: {
    read: (e) => Number.parseFloat(e),
    write: (e) => String(e)
  },
  any: {
    read: (e) => e,
    write: (e) => String(e)
  },
  string: {
    read: (e) => e,
    write: (e) => String(e)
  },
  map: {
    read: (e) => new Map(JSON.parse(e)),
    write: (e) => JSON.stringify(Array.from(e.entries()))
  },
  set: {
    read: (e) => new Set(JSON.parse(e)),
    write: (e) => JSON.stringify(Array.from(e))
  },
  date: {
    read: (e) => new Date(e),
    write: (e) => e.toISOString()
  }
}, cr = "vueuse-storage";
function Wu(e, o, r, t = {}) {
  var n;
  const {
    flush: i = "pre",
    deep: a = !0,
    listenToStorageChanges: l = !0,
    writeDefaults: s = !0,
    mergeDefaults: c = !1,
    shallow: d,
    window: u = he,
    eventFilter: g,
    onError: f = (v) => {
      console.error(v);
    }
  } = t, p = (d ? Gt : K)(o);
  if (!r)
    try {
      r = Dt("getDefaultStorage", () => {
        var v;
        return (v = he) == null ? void 0 : v.localStorage;
      })();
    } catch (v) {
      f(v);
    }
  if (!r)
    return p;
  const h = Oe(o), x = Ou(h), m = (n = t.serializer) != null ? n : Eu[x], { pause: y, resume: w } = Iu(p, () => z(p.value), { flush: i, deep: a, eventFilter: g });
  return u && l && (ir(u, "storage", k), ir(u, cr, B)), k(), p;
  function z(v) {
    try {
      if (v == null)
        r.removeItem(e);
      else {
        const H = m.write(v), M = r.getItem(e);
        M !== H && (r.setItem(e, H), u && u.dispatchEvent(new CustomEvent(cr, {
          detail: {
            key: e,
            oldValue: M,
            newValue: H,
            storageArea: r
          }
        })));
      }
    } catch (H) {
      f(H);
    }
  }
  function _(v) {
    const H = v ? v.newValue : r.getItem(e);
    if (H == null)
      return s && h !== null && r.setItem(e, m.write(h)), h;
    if (!v && c) {
      const M = m.read(H);
      return pu(c) ? c(M, h) : x === "object" && !Array.isArray(M) ? sr(sr({}, h), M) : M;
    } else
      return typeof H != "string" ? H : m.read(H);
  }
  function B(v) {
    k(v.detail);
  }
  function k(v) {
    if (!(v && v.storageArea !== r)) {
      if (v && v.key == null) {
        p.value = h;
        return;
      }
      if (!(v && v.key !== e)) {
        y();
        try {
          p.value = _(v);
        } catch (H) {
          f(H);
        } finally {
          v ? Pr(w) : w();
        }
      }
    }
  }
}
function Tt(e) {
  return _u("(prefers-color-scheme: dark)", e);
}
var Lu = Object.defineProperty, dr = Object.getOwnPropertySymbols, Ru = Object.prototype.hasOwnProperty, ju = Object.prototype.propertyIsEnumerable, pr = (e, o, r) => o in e ? Lu(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, Vu = (e, o) => {
  for (var r in o || (o = {}))
    Ru.call(o, r) && pr(e, r, o[r]);
  if (dr)
    for (var r of dr(o))
      ju.call(o, r) && pr(e, r, o[r]);
  return e;
};
function Nu(e = {}) {
  const {
    selector: o = "html",
    attribute: r = "class",
    initialValue: t = "auto",
    window: n = he,
    storage: i,
    storageKey: a = "vueuse-color-scheme",
    listenToStorageChanges: l = !0,
    storageRef: s,
    emitAuto: c
  } = e, d = Vu({
    auto: "",
    light: "light",
    dark: "dark"
  }, e.modes || {}), u = Tt({ window: n }), g = F(() => u.value ? "dark" : "light"), f = s || (a == null ? K(t) : Wu(a, t, i, { window: n, listenToStorageChanges: l })), p = F({
    get() {
      return f.value === "auto" && !c ? g.value : f.value;
    },
    set(y) {
      f.value = y;
    }
  }), h = Dt("updateHTMLAttrs", (y, w, z) => {
    const _ = n == null ? void 0 : n.document.querySelector(y);
    if (_)
      if (w === "class") {
        const B = z.split(/\s/g);
        Object.values(d).flatMap((k) => (k || "").split(/\s/g)).filter(Boolean).forEach((k) => {
          B.includes(k) ? _.classList.add(k) : _.classList.remove(k);
        });
      } else
        _.setAttribute(w, z);
  });
  function x(y) {
    var w;
    const z = y === "auto" ? g.value : y;
    h(o, r, (w = d[z]) != null ? w : z);
  }
  function m(y) {
    e.onChanged ? e.onChanged(y, x) : x(y);
  }
  return _e(p, m, { flush: "post", immediate: !0 }), c && _e(g, () => m(p.value), { flush: "post" }), Pt(() => m(p.value)), p;
}
var Gu = Object.defineProperty, Uu = Object.defineProperties, qu = Object.getOwnPropertyDescriptors, ur = Object.getOwnPropertySymbols, Ku = Object.prototype.hasOwnProperty, Qu = Object.prototype.propertyIsEnumerable, fr = (e, o, r) => o in e ? Gu(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, Ju = (e, o) => {
  for (var r in o || (o = {}))
    Ku.call(o, r) && fr(e, r, o[r]);
  if (ur)
    for (var r of ur(o))
      Qu.call(o, r) && fr(e, r, o[r]);
  return e;
}, Yu = (e, o) => Uu(e, qu(o));
function Xu(e = {}) {
  const {
    valueDark: o = "dark",
    valueLight: r = "",
    window: t = he
  } = e, n = Nu(Yu(Ju({}, e), {
    onChanged: (l, s) => {
      var c;
      e.onChanged ? (c = e.onChanged) == null || c.call(e, l === "dark") : s(l);
    },
    modes: {
      dark: o,
      light: r
    }
  })), i = Tt({ window: t });
  return F({
    get() {
      return n.value === "dark";
    },
    set(l) {
      l === i.value ? n.value = "auto" : n.value = l ? "dark" : "light";
    }
  });
}
var gr;
(function(e) {
  e.UP = "UP", e.RIGHT = "RIGHT", e.DOWN = "DOWN", e.LEFT = "LEFT", e.NONE = "NONE";
})(gr || (gr = {}));
var Zu = Object.defineProperty, mr = Object.getOwnPropertySymbols, ef = Object.prototype.hasOwnProperty, of = Object.prototype.propertyIsEnumerable, Cr = (e, o, r) => o in e ? Zu(e, o, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[o] = r, rf = (e, o) => {
  for (var r in o || (o = {}))
    ef.call(o, r) && Cr(e, r, o[r]);
  if (mr)
    for (var r of mr(o))
      of.call(o, r) && Cr(e, r, o[r]);
  return e;
};
const tf = {
  easeInSine: [0.12, 0, 0.39, 0],
  easeOutSine: [0.61, 1, 0.88, 1],
  easeInOutSine: [0.37, 0, 0.63, 1],
  easeInQuad: [0.11, 0, 0.5, 0],
  easeOutQuad: [0.5, 1, 0.89, 1],
  easeInOutQuad: [0.45, 0, 0.55, 1],
  easeInCubic: [0.32, 0, 0.67, 0],
  easeOutCubic: [0.33, 1, 0.68, 1],
  easeInOutCubic: [0.65, 0, 0.35, 1],
  easeInQuart: [0.5, 0, 0.75, 0],
  easeOutQuart: [0.25, 1, 0.5, 1],
  easeInOutQuart: [0.76, 0, 0.24, 1],
  easeInQuint: [0.64, 0, 0.78, 0],
  easeOutQuint: [0.22, 1, 0.36, 1],
  easeInOutQuint: [0.83, 0, 0.17, 1],
  easeInExpo: [0.7, 0, 0.84, 0],
  easeOutExpo: [0.16, 1, 0.3, 1],
  easeInOutExpo: [0.87, 0, 0.13, 1],
  easeInCirc: [0.55, 0, 1, 0.45],
  easeOutCirc: [0, 0.55, 0.45, 1],
  easeInOutCirc: [0.85, 0, 0.15, 1],
  easeInBack: [0.36, 0, 0.66, -0.56],
  easeOutBack: [0.34, 1.56, 0.64, 1],
  easeInOutBack: [0.68, -0.6, 0.32, 1.6]
};
rf({
  linear: Cu
}, tf);
const Ce = Xu(), hr = xu(Ce), Cf = F(() => Ce.value ? du : null), nf = {
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  viewBox: "0 0 24 24"
}, af = /* @__PURE__ */ Hr(
  "path",
  {
    d: "M6.76 4.84l-1.8-1.79l-1.41 1.41l1.79 1.79zM1 10.5h3v2H1zM11 .55h2V3.5h-2zm8.04 2.495l1.408 1.407l-1.79 1.79l-1.407-1.408zm-1.8 15.115l1.79 1.8l1.41-1.41l-1.8-1.79zM20 10.5h3v2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6s6-2.69 6-6s-2.69-6-6-6zm0 10c-2.21 0-4-1.79-4-4s1.79-4 4-4s4 1.79 4 4s-1.79 4-4 4zm-1 4h2v2.95h-2zm-7.45-.96l1.41 1.41l1.79-1.8l-1.41-1.41z",
    fill: "currentColor"
  },
  null,
  -1
  /* HOISTED */
), lf = [af], It = te({
  name: "WbSunnyOutlined",
  render: function(o, r) {
    return me(), wr("svg", nf, lf);
  }
}), sf = {
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  viewBox: "0 0 24 24"
}, cf = /* @__PURE__ */ Hr(
  "path",
  {
    d: "M9.37 5.51A7.35 7.35 0 0 0 9.1 7.5c0 4.08 3.32 7.4 7.4 7.4c.68 0 1.35-.09 1.99-.27A7.014 7.014 0 0 1 12 19c-3.86 0-7-3.14-7-7c0-2.93 1.81-5.45 4.37-6.49zM12 3a9 9 0 1 0 9 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 0 1-4.4 2.26a5.403 5.403 0 0 1-3.14-9.8c-.44-.06-.9-.1-1.36-.1z",
    fill: "currentColor"
  },
  null,
  -1
  /* HOISTED */
), df = [cf], zt = te({
  name: "DarkModeOutlined",
  render: function(o, r) {
    return me(), wr("svg", sf, df);
  }
}), hf = /* @__PURE__ */ te({
  __name: "switch",
  props: {
    round: { type: Boolean, default: !1 },
    darkBackground: { default: "linear-gradient(135deg,#42e695,#3bb2b8)" },
    lightBackground: { default: "" }
  },
  setup(e) {
    const o = e, r = ({ checked: t }) => ({
      background: t ? o.darkBackground : o.lightBackground
    });
    return (t, n) => (me(), je(N(au), {
      round: t.round,
      "rail-style": r,
      value: N(Ce),
      "onUpdate:value": n[0] || (n[0] = (i) => $r(Ce) ? Ce.value = i : null)
    }, {
      "checked-icon": ie(() => [
        ge(N(Je), null, {
          default: ie(() => [
            Ne(t.$slots, "dark", {}, () => [
              ge(N(zt))
            ])
          ]),
          _: 3
        })
      ]),
      "unchecked-icon": ie(() => [
        ge(N(Je), null, {
          default: ie(() => [
            Ne(t.$slots, "light", {}, () => [
              ge(N(It))
            ])
          ]),
          _: 3
        })
      ]),
      _: 3
    }, 8, ["round", "value"]));
  }
}), pf = /* @__PURE__ */ te({
  __name: "switch-icon",
  props: {
    size: { default: "20px" }
  },
  setup(e) {
    return (o, r) => (me(), je(Sr, {
      name: "rotate",
      mode: "out-in"
    }, {
      default: ie(() => [
        N(Ce) ? (me(), je(N(Je), {
          key: 0,
          size: o.size,
          class: "cursor-pointer flex items-center",
          onClick: r[0] || (r[0] = (t) => N(hr)(!1))
        }, {
          default: ie(() => [
            Ne(o.$slots, "dark", {}, () => [
              ge(N(zt))
            ], !0)
          ]),
          _: 3
        }, 8, ["size"])) : (me(), je(N(Je), {
          key: 1,
          size: o.size,
          class: "cursor-pointer flex items-center",
          onClick: r[1] || (r[1] = (t) => N(hr)(!0))
        }, {
          default: ie(() => [
            Ne(o.$slots, "light", {}, () => [
              ge(N(It))
            ], !0)
          ]),
          _: 3
        }, 8, ["size"]))
      ]),
      _: 3
    }));
  }
});
const uf = (e, o) => {
  const r = e.__vccOpts || e;
  for (const [t, n] of o)
    r[t] = n;
  return r;
}, xf = /* @__PURE__ */ uf(pf, [["__scopeId", "data-v-43e699cf"]]);
export {
  hf as Switch,
  xf as SwitchIcon,
  Ce as isDark,
  Cf as naiveTheme,
  hr as toggleDark
};
